<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 --><head><!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"--><!-- #BeginEditable "doctitle" --> <title>Molecular Data</title><!-- #EndEditable --><meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"></head><body bgcolor="#FFFFFF"><div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%">   <table border="0" cellspacing="0" cellpadding="0" width="100%">    <tr> 	  <td align="left" valign="top" height="60">         <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">          <tr>        		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="../images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>           <td bgcolor="#FFFFFF">               <div align="left"><font face="Trebuchet MS"><img src="../images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></font></div>            </td>          </tr>			<tr> 			  <td height="42" align="left" valign="top"></td>			  <td height="42" align="left" valign="top"><font face="Trebuchet MS"><img src="../images/mesquiteLower.gif"></font></td>			</tr>        </table>        <a href="../manual.html"><img src="../images/manual.jpg" width="100" height="29" border="0"></a></td>	<td>        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">          <tr valign="top" height="92">            	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="../images/fadeRight.gif" width="72" height="160"></font></td>			<td bgcolor="#99FF99"><img src="../images/introduction.gif" valign="bottom"></td>			<td align="center" valign="top" height="92" width = "160" bgcolor="#99FF99">		  		<div align="left">                 <p><font face="Trebuchet MS"><span class="smallplain"><a href="../manual.html"><br>                  Introduction<br>                  </a></span><a href="../whyMesquite.html">Why?</a><br>                  <a href="../publication.html">Publication</a><br>                  <a href="../support.html">Support</a><br>                  <a href="../credits.html">Credits</a> </font><br>                  <font face="Trebuchet MS"><span class="smallplain"><a href="../help.html">Help                   </a></span></font><br>                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web                   Site</font></a></p>              </div>			</td>      		<td bgcolor="#99FF99"><img src="../images/basics.gif" valign="bottom"></td>			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99">               <p><font face="Trebuchet MS"><span class="smallplain"><a href="../help.html"><br>                </a><a href="../managingFiles.html">Files</a></span> <br>                <span class="smallplain"><a href="../menus.html">Menus</a><br>                <a href="../windows.html">Windows</a><br>				<a href="../charts.html">Charts</a><br>                <a href="../scripting.html">Scripts/Macros</a> <br>                <a href="../modules.html">Modules</a><br>				<a href="../howMesquiteWorks.html">How</a></span></font></p>              </td>      		<td bgcolor="#99FF99"><img src="../images/objects.gif" valign="bottom"></td>			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">				<p><font face="Trebuchet MS"><br>                <span class="smallplain"><a href="../characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="../Taxa.html"><br>                Taxa<br>                </a></span></font><font face="Trebuchet MS"><a href="../trees.html">Trees</a><br>                <a href="../glossary.html">Glossary</a> </font></p>              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="../changes.html">New</a><br>                &nbsp;&nbsp;<a href="../changes.html">Features</a></font></em></font></p>              </td>      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="../images/fadeLeft.gif" width="92" height="160"></font></td>		</tr>    	<tr>		    <td width="72" height="20"></td>		<td></td>		    <td>&nbsp;</td>            <td>&nbsp;</td>            <td>&nbsp;</td>		</tr>		</table>		</td>	</tr>  </table>  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">    <tr>       <td valign="top" align="left" width="110" height="100%">         <table border="0" cellspacing="0" cellpadding="0" width="142">          <tr>             <td width="6" bgcolor="#99FF99"><font face="Trebuchet MS"><img src="../images/fadeDown.gif" width="5" valign = "top" height="95" border="0" alt=" "></font></td>            <td colspan="2" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="../images/fadeDown.gif" width="100%" height="72" border="0" alt=" "><img src="../images/analyses.gif"></font></p>              </td>          </tr>          <tr bgcolor="#99FF99">             <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>            <td width="131" height="35" align="left" valign="middle"><p><font face="Trebuchet MS"><span class="smallplain"><a href="../CharacterEvolution/charEvolution.html">Character</a><br>                &nbsp; <a href="../CharacterEvolution/charEvolution.html">Evolution</a></span> <br>                <span class="smallplain"><br>                <a href="../RandomSimul/RandomSimul.html">Simulations &amp;</a><br>                &nbsp; <a href="../RandomSimul/RandomSimul.html">Randomizations</a></span>                 <br>                <br>                <a href="../popGen/popGen.html">Population</a><br>                &nbsp; <a href="../popGen/popGen.html">Genetics</a>				<br>                <br>                <a href="molecular.html">Molecular</a><br>&nbsp; <a href="molecular.html">Data</a></font></p>              <p><font face="Trebuchet MS"><a href="../continuous/continuous.html">Continuous</a><br>                &nbsp; <a href="../continuous/continuous.html">Characters</a></span></font></p>              <p><font face="Trebuchet MS"><br>                <a href="../studies/index.html">Studies</a></font></p></td>            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>          </tr>          <tr bgcolor="#99FF99">             <td height="35" align="left" valign="middle">&nbsp;</td>            <td height="35" align="left" valign="middle">&nbsp;</td>            <td height="35" align="left" valign="middle">&nbsp;</td>          </tr>          <tr>             <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="5" height="92"></font></td>            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="100%" height="92"></font></td>          </tr>        </table>      </td>      <td valign="top" width="100%" height="100%">         <table width="700" border="0" cellpadding="10" cellspacing="0">          <tr>             <td><font face="Trebuchet MS"><!-- #BeginEditable "body" -->               <h2>Molecular data</h2>              <p>Molecular data (DNA or protein sequences) can be edited, manipulated,                 simulated and analyzed in various ways in Mesquite. Most of the                 features discussed elsewhere concerning editing and analysis of                 general categorical data also apply to molecular data; here we                 focus on features specifically designed for sequence data.</p>              <h4>Contents</h4>              <ul>                <li><a href="#editing">Editing molecular data</a></li>                <li><a href="#simDNA">Simulating DNA sequence evolution</a></li>                <li><a href="#statsDNA">Statistics for DNA sequences</a></li>                <li><a href="#statsProtein">Statistics for Protein sequences</a></li>                <li><a href="#tertiary">Visualizing tertiary structure</a></li>                <li><a href="#popGen">Sequence Data within populations</a></li>                <li><a href="#rec">Reconstructing ancestral states</a></li>              </ul>              <p align="center"><img src="../images/DNAMatrix.gif" width="437" height="200"></p>              <h2><a name="editing" id="editing"></a>Editing molecular data</h2>              <p>Molecular data can be imported from files of NBRF format, PHYLIP                 format, and simple table format. It can also be exported to these                 formats.</p>              <p>The <a href="../characters.html#editor">Character Matrix Editor</a>                 can be used to edit a molecular sequence matrix. Standard ambiguity                 codes are allowed.</p>              <p>The following can be applied to all or the selected portions                 of a molecular sequence matrix in the Character Matrix Editor.                 These are available under the Alter/Transform submenu of the Matrix                 menu:</p>              <ul>                <li><strong>Nucleotide complement</strong> (DNA matrix only) &#8212;                   enters the complementar sequence into the selected cells</li>                <li><strong>Reverse sequence</strong> &#8212; reverses the order                   of contiguously selected blocks of sequence</li>                <li><strong>Collapse Gaps</strong> &#8212; collapses gaps to yield                   unaligned sequences</li>                <li><strong>Collapse Gaps-Only Edges</strong> &#8212; deletes                   characters at edges of matrix that are gaps-only.</li>              </ul>              <p>Other options may appear; see the page on <a href="../characters.html">characters</a>                 for standard choices in this submenu. You can also apply the other                 editing tools described for <a href="../characters.html">character                 matrices</a>.</p>              <p>The view of the matrix can be adjusted in various ways. Cells                 can be colored according to the state at the site (<strong>Color                 Cells</strong> submenu, <strong>Character State</strong>) or according                 to a value like the GC bias (Color Cells submenu, <strong>Cell                 Value</strong>; can request this coloring to use a moving window).                 Examples of this are shown below. The Display submenu of the Matrix                 menu contains other options such as a <strong>Bird's eye view</strong>                 which makes the cells narrow to show more of the sequences.</p>              <p><strong>Copy Sequence</strong> (at bottom of Edit menu) copies                 the selected cells of the matrix into the computer's clipboard                 as a sequence. That is, whereas the standard Copy would place                 into the clipboard selected pieces of the matrix in tab-delimited                 text format (e.g., if the sequence AATCA is selected, &quot;A-tab-A-tab-T-tab-C-tab-A&quot;                 would be copied), this modified Copy Sequence command does not                 include tabs (thus, &quot;AATCA&quot; would be copied). This style                 of copying is useful when interacting with programs like <a href="http://www.genecodes.com">Sequencher                 (TM)</a>. For instance, if you want to find a piece of sequence                 in a matrix in Mesquite within a chromatogram viewer of Sequencher,                 do the following: select the sequence in Mesquite, choose Copy                 Sequence, then go to Sequencher, select Find Bases, and paste                 the sequence as the search string.</p>              <p>Pieces of sequences can be found using the Find Sequence and                 Find All Sequences submenus of the Edit menu. The current options                 are:</p>              <ul>                <li> <strong>Matching Sequence</strong>: This finds sequences                   matching a target sequence the user enters. The search allows                   a certain number of mismatches. Optionally, it can search for                   the reverse, complement and reverse complement of the target                   sequence. </li>                <li><font color="#000000"> <strong>Maintain Target Match</strong>:                   This highlights and maintains highlighted the first occurence                   of a given sequence in a given taxon. First, you are asked which                   taxon to search in. Then, it displays a panel like this:<br>                  <img src="../images/maintainTargetMatch.jpg" width="350" height="27"><br>                  underneath the matrix. The first button (red X) is to close                   the panel; the second pauses the search; the third allows you                   to select another taxon as your focus. If you type a sequence                   into the text area, the matching sequence (if any) will be highlighted                   in the matrix. Mesquite is constantly monitoring this text,                   and so you don't need to give any command to find again if you                   change the text. This is useful if working with a program like                   Sequencher. If you see a stretch of sequence while viewing chromatograms                   that you'd like to find in the matrix in Mesquite, type in the                   sequence into the text box and you will quickly be taken to                   it in the taxon.</font></li>                <li><font color="#000000"><strong>Maintain Clipboard Match</strong>:                   This is similar to Maintain Target Match, except that it obtains                   the search string not from the text area but from the clipboard.                   If the clipboard changes, the function will automatically find                   the sequence again in the matrix. This is useful if working                   with a program like Sequencher. If you turn on Maintain Clipboard                   Match, then you can copy stretches of a sequence within Sequencher,                   and Mesquite will automatically highlight it, without your having                   to return to Mesquite or give any other command to it. (Mesquite                   is constantly monitoring the clipboard to see if it changes).</font></li>              </ul>              <p>&nbsp; </p>              <font face="Trebuchet MS">               <h2><a name="simDNA" id="simDNA"></a>Simulating DNA sequence evolution</h2>              <p>DNA sequence evolution can be simulated to build statistical                 tests, for instance via parametric bootstrapping. See the page                 on <a href="../RandomSimul/simDNA.html">simulating DNA sequences</a>.</p>              </font>               <h2><a name="statsDNA" id="statsDNA"></a>Statistics for DNA sequences</h2>              <p>Calculations for categorical characters in general can be applied                 to DNA sequences. For example, <a href="../CharacterEvolution/AncestralStates.html#parsimony">Parsimony                 calculations</a> can be made for DNA sequences, as can basic descriptive                 statistics such as the percent of a sequence or character that                 is missing data or gaps. In addition, there are several modules                 specifically designed for DNA data, illustrated by examples in                 Mesquite_Folder/examples/Molecular. These calculate compositional                 bias:</p>              <ul>                <li><strong>ACGT Compositional Bias</strong> &#8212; This module                   supplies the compositional bias of taxa, measured over the taxon's                   sequence. The bias is treated as a continuous character, and                   thus can be used wherever characters are used, as for instance                   in the reconstruction of the evolution of compositional bias                   as shown in the image below. It can return either the proportion                   G+C, or separately A, C, G, and T proportions.</li>              </ul>              <p align="center"><img src="images/gcBiasTaxa.jpg" width="446" height="350"></p>              <ul>                <li><strong>Character Compositional Bias</strong> &#8212; This                   module supplies the compositional bias for characters. It calculates                   the percent of taxa with particular nucleotides (GC bias, or                   individual frequency of A, C, G or T) for a character. The image                   below shows a moving window analysis of compositional bias along                   a sequence; the instructions for generating the chart are given                   <a href="../charts.html#gcbias">here</a>.</li>              </ul>              <p align="center"><img src="../images/gcSequence.jpg" width="557" height="188"></p>              <ul>                <li><strong>GC bias coloring of matrices</strong> &#8212; The                   cells of the Character Matrix Editor may be colored according                   to a moving window of GC bias along the sequence, as shown below,                   by selecting <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Color                   Cells&gt;Color By Cell Value</font></u>, then once shown the                   colors can be smoothed by a moving window analysis by selecting                   <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Moving                   Window (for colors)</font></u>.</li>              </ul>              <p align="center"><img src="images/gcBiasColorMatrix.jpg" width="371" height="263"></p>              <h2>Statistics for Protein Data</h2>              <p>               <ul>                <li><strong>Site hydrophobicity</strong> &#8212; This module supplies                   the average amino acid hydrophobicity, averaged across taxa,                   for each site. It can be used in charts, for instance to see                   the relationship between a phylogenetic statistic for the site                   (character) and it average hydrophobicity. This <a href="../charts.html#varhydrophob">chart</a>,                   for example, shows parsimony character steps as a function of                   hydrophobicity:</li>              </ul></p>              <p align="center"><img src="../images/stepsVsHydro.jpg" width="260" height="217"></p>              <p>               <ul>                <li><strong>Amino Acid hydrophobicity</strong> &#8212; The cells                   of the Character Matrix Editor may be colored according to a                   moving window of hydrophobicity along the sequence, as shown                   below, by selecting <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Color                   Cells&gt;Color By Cell Value</font></u>, then once shown the                   colors can be smoothed by a moving window analysis by selecting                   <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Moving                   Window (for colors)</font></u>.</li>              </ul></p>              <p align="center"><img src="images/aahydroColorMatrix.jpg" width="588" height="212"></p>              <h2><a name="tertiary"></a>Visualizing tertiary structure</h2>              <p>Although there are not yet dedicated windows for visualizing                 phylogenetic statistics in the context of molecular structure,                 features have been added to the Scattergram chart to allow it                 to be adapted for this purpose. For instance, in this image cytochrome                 B is shown, with the amino acids colored according to a simple                 phylogenetic statistic: the number of parsimony steps on a phylogeny.                 The colors are smoothed by a moving window, and show that several                 coils of the molecule, a few at the left and one deep at the right,                 evolve more rapidly than others. This example is illustrated in                 the data file at Mesquite_Folder/examples/Molecular/06-cytochromeB.nex</p>              <p align="center"><img src="images/cytStructure.jpg" width="360" height="292"></p>              <p>To build such a chart, begin with a file with a matrix of protein                 sequences. The procedure is also described in the example files                 08-cytochromeBlinked.nex and 09-cytochromeBscatter.nex. </p>              <p>               <ul>                <li>Select <strong>New Linked Matrix</strong> from the Characters                   menu. When a matrix is made to be linked to a second matrix,                   the two matrices are constrained to have the same number of                   characters. </li>                <li>Indicate that you want the linked matrix to be a Continuous                   matrix, and link it to your protein matrix. Then, turn it into                   a three dimensional matrix (Taxa X Characters X Coordinates                   [x, y and z]) by using <strong>Add Item</strong> and <strong>Rename                   Item</strong> in the Utilities submenu of the Matrix menu of                   the Character Matrix Editor. The x,y,z coordinates could be                   added for all taxa if known, but otherwise only one taxon needs                   to be filled out (because we will use the average x,y,z coordinates                   for the amino acids).</li>                <li>Once the linked matrix of xyz amino acid positions is entered,                   select <u><font size="-1" face="Arial, Helvetica, sans-serif">Analysis&gt;New                   Scattergram for&gt; Characters</font></u>. Indicate you want                   the scattergram to be for <strong>Stored Characters</strong>,                   and indicate <strong>Same</strong> value for the two axes. In                   the dialog box &quot;Values for axes&quot;, choose <strong>Mean                   Value of Character (Linked Matrix)</strong>. In response to                   &quot;Use characters from which matrix? (for Character Source)&quot;                   choose the protein sequence matrix as the matrix to be used.                   This will plot the sites (amino acids, characters) in their                   correct places, but as a series of round spots.</li>                <li>To change the appearance of the plot, select <strong>Join                   the Dots</strong> in the Special Effects submenu of the Scattergram                   menu. Then select <strong>Thick Joints</strong>, deselect <strong>Show                   Dots</strong>, deselect <strong>Join First to Last</strong>,                   and set the marker size larger (e.g., 8). This will result in                   a plot as shown above, but without the colors.</li>                <li>Next, choose <strong>Color by Third Value</strong> from the                   Colors menu and choose the value by which to color the amino                   acids. For parsimony steps, for instance, choose <strong>Character                   Value with current tree</strong>.</li>                <li>Finally, to use a moving window to smooth the colors, select                   <strong>Moving Window for Colors</strong> from the Colors menu                   and indicate the window size (e.g., 5).</li>              </ul></p>              <h2><a name="popGen"></a>Sequence data within populations</h2>              <p>See the page on <a href="../popGen/popGen.html">population genetics</a>.</p>              <h2><a name="rec" id="rec"></a>Reconstructing ancestral states</h2>              <p>Ancestral states of continuous characters can be reconstructed                 as described in the page on <a href="../CharacterEvolution/AncestralStates.html">reconstructing                 ancestral states</a>. Likelihood methods are not yet available                 for molecular characters.</p></li>              </ul> <!-- #EndEditable --></font></td>          </tr>        </table>        <table border="0" cellspacing="0" cellpadding="5" width="100%">          <tr>             <td align="left" valign="top">               <hr>              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite              Home Page</font></a><font size="2"> | <a href="../manual.html">Mesquite              Manual</a></font></span></font></p>              <p><font face="Trebuchet MS"><span class="smallplain">Copyright                 &copy; 2002-2005 by <a href="http://salticidae.org/wpm/home.html">Wayne                 P. Maddison</a> and <a href="http://david.bembidion.org">David                 R. Maddison</a>.<br>&nbsp;All rights reserved.</span></font></p></td>          </tr>        </table>      </td>    </tr>  </table></div><font face="Trebuchet MS"> </font></body><!-- #EndTemplate --></html>