<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN""http://www.w3.org/TR/html4/loose.dtd"><html><!-- InstanceBegin template="/Templates/standard.dwt" codeOutsideHTMLIsLocked="false" --><head><meta http-equiv="Content-Type" content="text/html; charset="><!-- InstanceBeginEditable name="doctitle" -->
<title>Chromaseq Overview</title>
<!-- InstanceEndEditable --><link rel="stylesheet" href="styles/mesquitePackage.css" type="text/css"><!-- InstanceBeginEditable name="head" --><!-- InstanceEndEditable --></head><body><div id="masthead">  <div id="banner"><img src="splash.gif" width="164" height="64"></div>  <div id="mainTitle"><font id="packageName"><b>Chromaseq</b></font><BR><font id="version">Test Version 0.9<br>30 October 2005</font><BR>         David R. Maddison and Wayne P. Maddison</div>  <div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->Overview<!-- InstanceEndEditable --></div> </div><div><!-- end masthead --><div id="content"><!-- InstanceBeginEditable name="content" -->
  <p>Chromaseq will take a directory of chromatograms, shown on the left, below, and examine the names of the files contained therein. In the file name it will find the code for the sample, as well as the name of the primer. It will use the primer name to determine the name of the gene fragment, and will reorganize the directory into subdirectories, one for each gene fragment, and within those, one for each sample, as shown on the right, below. </p>
  <table width="200" border="0" align="center" cellspacing="6">
    <tr>
      <td><div align="center"><img src="images/ppBefore.jpg" alt="Directory Before Phred Phrap" width="246" height="326"></div></td>
      <td><div align="center"><img src="images/ppMiddle.jpg" alt="During Phred Phrap" width="276" height="301" align="bottom"></div></td>
    </tr>
    <tr>
      <td><div align="center">Before Chromaseq processing </div></td>
      <td><div align="center">After Chromaseq has reorganized files </div></td>
    </tr>
  </table>
  <p>Chromaseq also creates a text file containing scripts that are then processed by the operating system. The primary purpose is to start up the program Phred, and send it commands to make base calls on each chromatogram and judge the quality of each base call. The file also starts up the program Phrap, and sends it commands to assemble the reads into contigs. Here's some of the output from this step:</p>
  <p align="center"><img src="images/terminal.jpg" alt="Terminal with Phred Phrap" width="497" height="281"></p>
  <p>After Phred and Phrap have finished, Chromaseq then examines the files created, including the .ace file containing the contigs, and modifies the files, including adjusting base calls in the contigs if double peaks are present in higher-quality reads. It also moves the FASTA files produced to a separate directory:</p>
  <p align="center"><img src="images/ppAfter.jpg" alt="After Phred Phrap" width="278" height="422"></p>
  <p>If so chosen, Mesquite's Chromaseq will also import the sequences into a Mesquite project, along with the quality scores as judged by Phred and Phrap, and links to the chromatograms themselves. By default, the data will be presented as character matrices with cells colored by the Phred/Phrap quality scores (with white being very high quality, darker green being lower quality, and blue being lowest quality), and as a bird's eye view:</p>
  <p align="center"><img src="images/importBirdsEye.jpg" alt="Bird's Eye View of Imported Sequences" width="664" height="99"></p>
  <p>Of course, these (currently unaligned) sequences can also be viewed in more standard ways:</p>
  <p align="center"><img src="images/importNormal.jpg" alt="Imported Sequences, Normal View" width="784" height="102"></p>
  <p>As the links to the chromatogram files are preserved, Chromaseq can present a view of the chromatograms for each sequence: </p>
  <p align="center"><img src="images/viewer.jpg" alt="Chromatogram Viewer" width="704" height="516"></p>
  <p>In this window, base qualities can be shown for each read and for the consensus sequence, and conflicts are indicated. Bases calls can also be adjusted in this window, with those edited bases  transferred to Mesquite's data matrix. </p>
  <h2><a name="isNot"></a>What Chromaseq Is Not</h2>
  <p>Chromaseq is <em>not</em> intended as a full-featured replacement to programs like <a href="http://www.genecodes.com">Sequencher</a>&#8482; or <a href="http://www.codoncode.com">Aligner</a>&#8482;. Chromaseq was built by us for our personal needs in systematics. We obtain many sequences of a few genes from beetles and spiders; many of these sequences are high-quality, and don't require much processing other than assembly of the reads and trimming. However, the tedium of processing these many sequences, and the lack of integration of existing tools within a systematics framework, led us to develop a solution that worked for us. While it has many custom features for dealing with the sorts of sequence data one acquires typically in systematics, it does not have the many features one would want for a full chromatogram analysis package, including for handling difficult data.  </p>
<!-- InstanceEndEditable --></div><!--end content --><div id="navBar">  <div id="sectionLinks">    <ul>      <li><a href="index.html">Home</a></li>      <li><a href="overview.html">Overview</a></li>      <li><a href="quickstart.html">Quickstart</a></li>	  <li><a href="phredphrap.html">Phred / Phrap with Chromaseq</a></li>	  <ul id="subUL">	  	<li><a href="installingPP.html">Installing Phred &amp; Phrap </a></li>  		<li><a href="primers.html">Primer File</a></li>  		<li><a href="sampleCode.html">Sample Codes &amp; Names File</a></li>  		<li><a href="runningPP.html">Running Phred/Phrap</a></li>	  	<li><a href="namingRules.html">Chromatogram File Naming Rules </a></li>		<li><a href="quality.html">Quality Scores</a></li>		<li><a href="postPhrap.html">Post-Phrap Options</a></li>	  </ul>      <li><a href="viewer.html">Chromatogram Viewer</a></li>  		<ul id="subUL">	  		<li><a href="navigation.html">Navigation</a></li>  			<li><a href="display.html">Display Options</a></li>  			<li><a href="callingBases.html">Calling Bases</a></li>		</ul>      <li><a href="otherPrograms.html">Sequencher &amp; Others </a></li>      <li><a href="acknowledgements.html">Acknowledgements</a></li>      <li><a href="citation.html">Citation</a></li>      <li><a href="faq.html">FAQs</a></li>      <li><a href="future.html">Future Directions</a></li>      <li style="margin-top: 20px;"><a href="download.html">Download</a></li>      <li><a href="sourceCode.html">Source Code</a></li>    </ul>	<p>&nbsp;</p>	<ul><li> <a href="http://mesquiteproject.org"  style="border: 0px;"><img src="images/mesquiteIcon.gif" alt="Mesquite Home" width="32" height="32"> Mesquite Home</a> </li>    </ul>  </div></div></div><div id="navBarExtension">&nbsp;</div><!--end navbar --><div id="siteInfo"> Copyright &copy; 2005 <a href="http://david.bembidion.org">David R. Maddison</a> and <a href="http://salticidae.org/wpm/home.html">Wayne P. Maddison</a> </div><br></body><!-- InstanceEnd --></html>