<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN""http://www.w3.org/TR/html4/loose.dtd"><html><!-- InstanceBegin template="/Templates/standard.dwt" codeOutsideHTMLIsLocked="false" --><head><meta http-equiv="Content-Type" content="text/html; charset="><!-- InstanceBeginEditable name="doctitle" -->
<title>Chromaseq Quickstart</title>
<!-- InstanceEndEditable --><link rel="stylesheet" href="styles/mesquitePackage.css" type="text/css"><!-- InstanceBeginEditable name="head" -->
<style type="text/css">
<!--
.style1 {	font-family: "Courier New", Courier, monospace;
	font-size: 12px;
}
-->
</style>
<!-- InstanceEndEditable --></head><body><div id="masthead">  <div id="banner"><img src="splash.gif" width="164" height="64"></div>  <div id="mainTitle"><font id="packageName"><b>Chromaseq</b></font><BR><font id="version">Test Version 0.9<br>30 October 2005</font><BR>         David R. Maddison and Wayne P. Maddison</div>  <div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->Quickstart<!-- InstanceEndEditable --></div> </div><div><!-- end masthead --><div id="content"><!-- InstanceBeginEditable name="content" -->
  <h2>Preparing Phred, Phrap, and Phd2Fasta</h2>
  <p>Make sure Phred, Phrap, and Phd2Fasta are <a href="installingPP.html">compiled and installed</a> correctly, and the phredpar.dat file is <a href="installingPP.html#phredpar">edited correctly</a>. </p>
  <h2>Preparing the primer file</h2>
  <p>You will need to create a <a href="primers.html">primer file</a> documenting the gene fragments and a list of the names of their forward and reverse primers. </p>
  <h2>Preparing the sample names file</h2>
  <p>If you wish for your sequences to be named by something other than the sample codes contained in the file name, then you will need to create a text file that contains a list of the sample codes and the names to which they should be translated. See <a href="sampleCode.html">&quot;Sample names file&quot;</a> for a description of the format of this text file.  </p>
  <h2><a name="runningPhPh"></a>Assembling the sequences: Running Phred and Phrap </h2>
  <p>If you wish to ask Mesquite to organize the chromatogram files, have Phred and Phrap  assemble the contigs, and have Mesquite process the results, importing them into a new project, then choose <font id="menu">Open Other&gt;Phred/Phrap Import from Chromatograms...</font> from the <font id="menu">File</font> menu. </p>
  <p>If you wish to ask Mesquite to do this but add the processed sequences to an existing project, then with that project open, choose <font id="menu">Edit&gt;Append Sequences From Chromatograms...</font></p>
  <p>If you wish to ask Mesquite to do this but NOT import the sequences into a new or existing project, then choose <font id="menu">Utilites&gt;Run Phred and Phrap...</font> when the Mesquite Log window is the frontmost window.</p>
  <p>You will then be presented with a dialog box that looks something like this: </p>
  <p align="center"><img src="images/runPhredPhrapEmpty.jpg" alt="Run Phred Phrap Dialog" width="359" height="357"></p>
  <h3>Defining a naming rule</h3>
  <p>The first thing to do is define a Naming Rule, which describes how the chromatogram files are named. In particular, you need to give Mesquite enough details so that it can exactly find the sample code and primer name within the file name. For example, one of the example chromatograms has as a file name:</p>
  <p><span class="style1">&nbsp;&nbsp;&nbsp;A11_A11DNA0446_D3_340852.ab1</span></p>
  <p>In this name, the sample code is present immediately following the distinctive string of characters &quot;DNA&quot;. The sample code ends with an underscore (&quot;_&quot;). The primer name in this file name is &quot;D3&quot;; it is preceded by an underscore, and followed by an underscore. The rest of the name (except, if you choose, the extension &quot;.ab1&quot;) is irrelevant to Mesquite. If you touch on Edit Naming Rule, and then in the list dialog that appears touch on New and give your rule a name, you will see a dialog in which you can edit the naming rule: </p>
  <p align="center"><img src="images/nameRuleEditingDefault.jpg" alt="Name Rule Editing" width="249" height="295"></p>
  <p>The default naming rule is one that matches the file &quot;<span class="style1">A11_A11DNA0446_D3_340852.ab1</span>&quot;, as that's how David's files are named. </p>
  <p>Adjust your naming rule to match your file name structure. Once a naming rule is defined, it will be saved to disk and it will be an option for you in the future in the popup menu beside &quot;File Naming Rules&quot; in the Phred/Phrap dialog box. All file names in the directory must use the same naming rule. </p>
  <h3>Setting the locations of files</h3>
  <p>Use the Browse button in the Phred/Phrap dialog box to choose the location of the file containing the sample names file (if you don't to translate the sample codes, then you can leave this box blank, but you should then uncheck &quot;translate sample codes using names file&quot;). </p>
  <p>Use the Browse button to choose your primer  file. </p>
  <p>Touch on the Phred Phrap Locations &amp; Options button to set the locations of both the directory containing Phred, Phrap, and Phd2Fasta, as well as the location of the phredpar.dat file.</p>
  <h3>Setting other options</h3>
  <p>You can also set other options, including for the processing Mesquite does after Phred and Phrap examine the files. Explore <a href="postPhrap.html">Post-Phrap Options</a> for some things you might change. </p>
  <h3>Press OK</h3>
  <p>You are now ready to have Phred and Phrap run. Press the OK button in the Run Phred Phrap dialog box. Mesquite's log window should show the details of the processing. You might want to bring this frontmost so that you can watch it. In addition, the operating system shell or Terminal should start up and you should see Phred and Phrap produce lots of information that scrolls by. </p>
  <p>If you asked Mesquite to process your files and import the results into a Mesquite project, then you should see Mesquite open a character matrix editor and present you with the results.</p>
  <h2>Viewing the sequences in Mesquite's editor</h2>
  <p>Mesquite will present imported sequences organized with all sequences from one gene fragment together in one editor, with a different editor for each gene fragment. By default the editor will be shown in &quot;bird's eye view&quot; (as if from a distance, with each base occupying only a thin vertical strip), with the color of a base being related to the quality score for that base as judged by Phred and Phrap.</p>
  <p><img src="images/importInitial.jpg" alt="Imported Sequences" width="638" height="400"></p>
  <p> If you turn on the character matrix's Colors Legend (by touching on the <img src="images/showLegendButton.jpg" alt="Show Colors Legend Button" width="16" height="12"> button below the taxon names), you can see the meaning of the colors:</p>
  <p><img src="images/importLegend.jpg" alt="ImportedSequences with Legend" width="782" height="406"></p>
  <p>The color of the taxon name is in proportion to the number of high-quality bases in its sequence, with darker taxon names indicating fewer high-quality bases.</p>
  <h2>Viewing a sequence's chromatograms</h2>
  <p>To view the chromatograms underlying a sequence, touch on that sequence using the View Chromatogram Tool (<img src="images/chromTool.jpg" alt="View Chromatogram Tool" width="30" height="29" align="absmiddle">). A window will appear containing that sequence's chromatograms:</p>
  <p><img src="images/viewerQS.jpg" alt="Chromatogram Viewer" width="497" height="407"></p>
  <p>The upper panels show the chromatogram traces. Ones with names in red and a pale gray background are reverse-complemented. The base colors under the traces are shown using the same quality score coloring as calculated by Phred and used in the imported matrix. At the bottom are three versions of the consensus sequence: the top one, the untrimmed version as produced by Phred/Phrap and as modified by Mesquite; the middle one, the trimmed version as originally imported into Mesquite; the bottom version, or edited version as it appears in the matrix. The bottom two will be identical unless you have changed the base calls. The colors of the bases in the top two of these sequences also indicates the base quality scores as determined by Phrap. </p>
  <p>A few  graphical features of note are:</p>
  <ul>
    <li>Conflicts between reads are shown as pink or red bars in the &quot;Phred.Phrap.Mesquite&quot;
      sequence panel. If the reads that conflict are of low quality, the bars
      will be pink; if they are of high quality, they will be red. </li>
    <li>The brown bars above a base call and under a peak in each read indicate
      that that peak was the one used by Phrap as the primary peak on which the
      call for the consensus was based. </li>
    <li>You can show or hide various element by using the <font id="menu">Chromatograms&gt;Show</font>      submenu,
      including traces of  bases (e.g., hide the G trace). </li>
    <li>You can ask to have low-quality sections of reads dimmed so that they
      don't visually distract from the high-quality reads in those some regions. </li>
  </ul>
  <p>One navigational feature of note is:</p>
  <ul>
    <li>The left and right arrow keys can be used to move along the chromatograms.
      You can choose (using the <font id="menu">Chromatograms&gt;Arrow Keys</font> submenu) whether these
      move to the next base, the next base with a conflict, or the next low-quality
      base.</li>
  </ul>
  <h2>Changing base calls</h2>
  <p>Touching on a base in the bottom consensus sequence will select that base
    and allow you to edit it by typing in a new base:</p>
  <p align="center"><img src="images/baseCall.jpg" alt="Making a base call" width="86" height="361"></p>
  <p>Any base that has been so edited is marked by a black bar underneath it:</p>
  <p align="center"><img src="images/afterBaseCall.jpg" alt="after base call" width="86" height="102"> </p>
  <p>The new base call is directly transferred in the sequence in Mesquite's
    character matrix. </p>
  <p>&nbsp; </p>
<!-- InstanceEndEditable --></div><!--end content --><div id="navBar">  <div id="sectionLinks">    <ul>      <li><a href="index.html">Home</a></li>      <li><a href="overview.html">Overview</a></li>      <li><a href="quickstart.html">Quickstart</a></li>	  <li><a href="phredphrap.html">Phred / Phrap with Chromaseq</a></li>	  <ul id="subUL">	  	<li><a href="installingPP.html">Installing Phred &amp; Phrap </a></li>  		<li><a href="primers.html">Primer File</a></li>  		<li><a href="sampleCode.html">Sample Codes &amp; Names File</a></li>  		<li><a href="runningPP.html">Running Phred/Phrap</a></li>	  	<li><a href="namingRules.html">Chromatogram File Naming Rules </a></li>		<li><a href="quality.html">Quality Scores</a></li>		<li><a href="postPhrap.html">Post-Phrap Options</a></li>	  </ul>      <li><a href="viewer.html">Chromatogram Viewer</a></li>  		<ul id="subUL">	  		<li><a href="navigation.html">Navigation</a></li>  			<li><a href="display.html">Display Options</a></li>  			<li><a href="callingBases.html">Calling Bases</a></li>		</ul>      <li><a href="otherPrograms.html">Sequencher &amp; Others </a></li>      <li><a href="acknowledgements.html">Acknowledgements</a></li>      <li><a href="citation.html">Citation</a></li>      <li><a href="faq.html">FAQs</a></li>      <li><a href="future.html">Future Directions</a></li>      <li style="margin-top: 20px;"><a href="download.html">Download</a></li>      <li><a href="sourceCode.html">Source Code</a></li>    </ul>	<p>&nbsp;</p>	<ul><li> <a href="http://mesquiteproject.org"  style="border: 0px;"><img src="images/mesquiteIcon.gif" alt="Mesquite Home" width="32" height="32"> Mesquite Home</a> </li>    </ul>  </div></div></div><div id="navBarExtension">&nbsp;</div><!--end navbar --><div id="siteInfo"> Copyright &copy; 2005 <a href="http://david.bembidion.org">David R. Maddison</a> and <a href="http://salticidae.org/wpm/home.html">Wayne P. Maddison</a> </div><br></body><!-- InstanceEnd --></html>