<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 -->
<head>
<!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"-->
<!-- #BeginEditable "doctitle" --> <title>Testing monophyly of a group</title><!-- #EndEditable -->
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body bgcolor="#FFFFFF">
<div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%"> 
  <table border="0" cellspacing="0" cellpadding="0" width="100%">
    <tr> 
	  <td align="left" valign="top" height="60"> 
        <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">
          <tr> 
       		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="../../images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>
           <td bgcolor="#FFFFFF"> 
              <div align="left"><font face="Trebuchet MS"><img src="../../images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></font></div>
            </td>
          </tr>
			<tr> 
			  <td height="42" align="left" valign="top"></td>
			  <td height="42" align="left" valign="top"><font face="Trebuchet MS"><img src="../../images/mesquiteLower.gif"></font></td>
			</tr>
        </table>
        <a href="../../manual.html"><img src="../../images/manual.jpg" width="100" height="29" border="0"></a></td>
	<td>
        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">
          <tr valign="top" height="92"> 
           	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="../../images/fadeRight.gif" width="72" height="160"></font></td>
			<td bgcolor="#99FF99"><img src="../../images/introduction.gif" valign="bottom"></td>
			<td align="center" valign="top" height="92" width = "160" bgcolor="#99FF99">
		  		<div align="left"> 
                <p><font face="Trebuchet MS"><span class="smallplain"><a href="../../manual.html"><br>
                  Introduction<br>
                  </a></span><a href="../../whyMesquite.html">Why?</a><br>
                  <a href="../../publication.html">Publication</a><br>
                  <a href="../../support.html">Support</a><br>
                  <a href="../../credits.html">Credits</a> </font><br>
                  <font face="Trebuchet MS"><span class="smallplain"><a href="../../help.html">Help 
                  </a></span></font><br>
                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web 
                  Site</font></a></p>
              </div>
			</td>
      		<td bgcolor="#99FF99"><img src="../../images/basics.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99"> 
              <p><font face="Trebuchet MS"><span class="smallplain"><a href="../../help.html"><br>
                </a><a href="../../managingFiles.html">Files</a></span> <br>
                <span class="smallplain"><a href="../../menus.html">Menus</a><br>
                <a href="../../windows.html">Windows</a><br>
				<a href="../../charts.html">Charts</a><br>
                <a href="../../scripting.html">Scripts/Macros</a> <br>
                <a href="../../modules.html">Modules</a><br>
				<a href="../../howMesquiteWorks.html">How</a></span></font>
				<!-- Google CSE Search Box Begins  -->
				  <form id="searchbox_016202157087875381635:khljhzi-ps8" action="http://www.google.com/cse">
				    <input type="hidden" name="cx" value="016202157087875381635:khljhzi-ps8" />
				    <input type="hidden" name="cof" value="FORID:0" />
				    <input name="q" type="text" size="20" />
				    <input type="submit" name="sa" value="Search Mesquite" />
				  </form>
				  <script type="text/javascript" src="http://www.google.com/coop/cse/brand?form=searchbox_016202157087875381635%3Akhljhzi-ps8"></script>
				<!-- Google CSE Search Box Ends --></p>
              </td>
      		<td bgcolor="#99FF99"><img src="../../images/objects.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">
				<p><font face="Trebuchet MS"><br>
                <span class="smallplain"><a href="../../characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="../../Taxa.html"><br>
                Taxa<br>
                </a></span></font><font face="Trebuchet MS"><a href="../../trees.html">Trees</a><br>
                <a href="../../glossary.html">Glossary</a> </font></p>
              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="../../changes.html">New</a><br>
                &nbsp;&nbsp;<a href="../../changes.html">Features</a></font></em></font></p>
              </td>
      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="../../images/fadeLeft.gif" width="92" height="160"></font></td>
		</tr>
    	<tr>
		    <td width="72" height="20"></td>
		<td></td>
		    <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td>&nbsp;</td>
		</tr>
		</table>
		</td>
	</tr>
  </table>
  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">
    <tr> 
      <td valign="top" align="left" width="110" height="100%"> 
        <table border="0" cellspacing="0" cellpadding="0" width="142">
		  <tr>
		    <td width="6"></td>
			<td><a href="../../toc.html"><img src="../../images/toc.gif" width="100" height="50" border="0" ></a></td>
		  </tr>
          <tr> 
            <td width="6" bgcolor="#99FF99"><font face="Trebuchet MS"><img src="../../images/fadeDown.gif" width="5" valign = "top" height="95" border="0" alt=" "></font></td>
            <td colspan="2" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="../../images/fadeDown.gif" width="100%" height="72" border="0" alt=" "><img src="../../images/analyses.gif"></font></p>
              </td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>
            <td width="131" height="35" align="left" valign="middle"><p>
			<font face="Trebuchet MS">
				<a href="../../CharacterEvolution/charEvolution.html">Character</a><br>
                	&nbsp; <a href="../../CharacterEvolution/charEvolution.html">Evolution</a> <br> <br>
                <a href="../../RandomSimul/RandomSimul.html">Simulations &amp;</a><br>
                	&nbsp; <a href="../../RandomSimul/RandomSimul.html">Randomizations</a> <br> <br>
                <a href="../../popGen/popGen.html">Gene trees</a> <br>  <br>
                <a href="../../molecular/molecular.html">Molecular</a><br>
					&nbsp; <a href="../../molecular/molecular.html">Data</a><br><br>
				<a href="../../continuous/continuous.html">Continuous</a><br>
               	 &nbsp; <a href="../../continuous/continuous.html">Characters</a><br> <br>
                <a href="../../otherPrograms.html">Use with</a><br>&nbsp; <a href="../../otherPrograms.html">Other Programs</a> <br><br>
                <a href="../index.html">Studies</a> </font></p></td>
            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr> 
            <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="../../images/fadeUp.gif" width="5" height="92"></font></td>
            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="../../images/fadeUp.gif" width="100%" height="92"></font></td>
          </tr>
        </table>
      </td>
      <td valign="top" width="100%" height="100%"> 
        <table width="700" border="0" cellpadding="10" cellspacing="0">
          <tr> 
            <td><font face="Trebuchet MS"><!-- #BeginEditable "body" -->               <h2>Testing monophyly of a group of beetles</h2>              <p>David R. Maddison</p>              <h3 align="left">The question</h3>              <p align="left">There is an enigmatic group of terrestrial beetles                 called the Trachypachidae (left, below). These had traditionally                 been considered to be related to other terrestrial beetles in                 the suborder Adephaga, but later analyses of morphological data                 suggested (Bell, 1966; Hammond, 1979; Roughley, 1981; Ward, 1979)                 that they were instead closely related to some water beetles,                 the Dytiscoidea (right, below).</p>              <p align="center"><img src="images/trachypachid.jpg" width="225" height="300"><img src="images/noterid.jpg" width="186" height="300"><br>                A terrestrial trachypachid (<em>Trachypachus holmbergi</em>, left)                 <br>                and an aquatic dytiscoid (<em>Hydrocanthus</em> sp., right).</p>              <p align="left">Shull et al. (2001) report sequence data of the                 18S rRNA gene suggesting that trachypachids are not related to                 dytiscoids, but are instead related to some other terrestrial                 beetles. For example, the most parsimonious trees in one analysis                 have trachypachids with other terrestrial adephagans, and not                 with dytiscoids. Forcing trachypachids with dytiscoids increases                 the treelength by 9 steps. </p>              <p align="left">However, given the vagaries of the evolutionary                 process and phylogenetic inference, it is possible that even if                 trachypachids and dytiscoids are truly related, one might incorrectly                 infer that they are not, just by chance or because of difficulties                 such as long branch attraction (Felsenstein, 1978; Hendy &amp;                 Penny, 1989; Huelsenbeck, 1997). A statistical test could help                 us determine whether or not a difference of 9 steps between trees                 with trachypachids placed with dytiscoids and unconstrained trees                 is expected under the hypothesis that trachypachids and dytiscoids                 are related.</p>              <h3 align="left">A statistical test</h3>              <p align="left">Is 9 steps a significant difference in treelength                 in this context? Can we can reject the monophyly of trachypachids                 plus dytiscoids?</p>              <p align="left">To test the hypotheses that trachypachids are related                 to dytiscoids, we first need to flesh out its details of the hypothesis.                 It needs to be detailed enough to allow us to predict the observations                 we would expect to see if this hypothesis were true. First of                 all, we will need a detailed phylogenetic hypothesis. We could                 find the tree of highest likelihood for 18S rDNA under the constraint                 that trachypachids are with dytiscoids; for one matrix, this tree                 is as shown below, with trachypachids marked in green and dytiscoids                 in blue.</p>              <p align="center"><img src="images/tdtree.gif" width="387" height="257"></p>              <p align="left">Branch lengths of this tree and parameter values                 for a GTR + Gamma + Proportion Invariant (GTR+G+I) model of evolution                 are estimated by maximum likelihood using the 18S rDNA data. We                 thus have built a model that contains our best guess of the nature                 of evolution presuming that trachypachids and dytiscoids form                 a clade; the details of the model were established using the 18S                 rDNA data. </p>              <p align="left">We can now ask: if evolution occurred under a GTR+G+I                 model with the parameter values as inferred, up the tree shown                 above (with trachypachids and dytiscoids forming a clade), then                 what would we expect the difference in treelength to be between                 the most parsimonious trees constrained to have trachypachids                 with dytiscoids and the most parsimonious unconstrained trees?                 Would that difference in general be similar to the observed value,                 9? Or is 9 an unexpected value? Specifically, is 9 a value that                 we would expect to observe less than 0.05 of the time? If so,                 then we could reject the hypothesis that trachypachids are related                 to dytiscoids. </p>              <p align="left">This test is an example of the use of parametric                 bootstrapping (Huelsenbeck et al., 1995; Swofford et al., 1996;                 Goldman et al., 2000). It is almost identical to the monophyly                 test proposed by Huelsenbeck et al. (1996), except that they use                 a difference in likelihood rather than treelength as their test                 statistic. Treelength is used here for pedagogical reasons as                 it allows the reader to conduct the test quickly; the methods                 described below could easily be modified to use likelihood values                 instead. </p>              <h3 align="left">Building the statistical test</h3>              <p align="left">To conduct this test, in Mesquite open the example                 file &quot;<strong>study001.nex</strong>&quot;, in the folder                 Mesquite_Folder/docs/mesquite/studies/study001/. This file contains                 a matrix of 18S rDNA, and the tree of highest likelihood found                 in which trachypachids are with dytiscids, with branch lengths                 inferred from the data. It also contains a GTR+G+I model with                 parameters inferred by PAUP* using maximum likelihood and 18S                 rDNA. We will ask Mesquite to simulate 100 matrices, and then                 (using the batch file Mesquite produces) ask PAUP* to find the                 most parsimonious trees with trachypachids constrained to be with                 dytiscoids and the most parsimonious unconstrained trees, and                 write their treelengths to a scorefile. Mesquite will then read                 in the results and calculate the distribution of treelength differences,                 allowing us to determine if the observed value of 9 is unusual.</p>              <p align="left">The simulations can be done by choosing <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree                 Window) Analysis &gt; Batch Architect &gt; Export Matrices &amp;                 Batch Files...</font></u>. After selecting &quot;Simulated Matrices                 on Current Tree&quot; and &quot;Evolve DNA Characters&quot;, choose                 the GTR+G+I model. You will then be presented with the Export                 Matrices &amp; Batch Files dialog box. Give a base name for the                 matrices of &quot;TDTest&quot;, and do 100 replicates. We will                 use the template &quot;PAUPConstraintTestParsimony&quot;:</p>              <p align="center"><img src="images/treeTestDialog.gif" width="336" height="356"></p>              <p align="left">If you wish to look at the contents of this template,                 touch on &quot;Edit Templates&quot;, and in the Template Manager,                 select &quot;PAUPConstraintTestParsimony&quot;, and press &quot;View&quot;.                 (As this template is built-in, you can't edit it, only view it.)                 You will see some elements of the batch files that will be produced.               </p>              <p align="center"><img src="images/templateEditor1.gif" width="343" height="411"></p>              <p align="left">The complexity of this template needn't be of concern.                 If you are interested, details of how it works are presented in                 &quot; <a href="../../../..//mesquite/batchArch/aBatchArchIntro/batchtemplates.html">Design                 of Batch Templates</a>&quot;. Just press &quot;OK&quot; to get                 back to the Template Manager, and then press &quot;Done&quot;.               </p>              <p align="left">You are almost ready to have Mesquite simulate the                 matrices. However, the template we are using requires that we                 specify a tree to use as a constraint tree. In this case, the                 constraint tree is one with trachypachids with dytiscoids, but                 with no other structure. A tree like this is stored in the file                 under the name &quot;TD constraint&quot;. To choose it, touch                 the &quot;Choose Tree&quot; button, and select &quot;TD constraint&quot;.</p>              <p align="left">All the options have now been chosen. Press the                 &quot;OK&quot; button in the Export Matrices and Batch Files dialog                 box to start the simulation. You will be asked for a location                 to save the files, and be given some messages. When you are asked                 for the number of characters, Mesquite recognizes that you are                 calculating some elements of the model using an existing matrix                 (in particular, the frequencies of A, C, G, and T), and for this                 reason it gives as the default number of characters the number                 in the original matrix. That's the number we want to use in this                 case, as we want the simulation to be as realistic as possible.</p>              <p align="left">Once the simulations are all done, then go into                 PAUP*, and execute the file "PaupCommands.nex". When PAUP* completes                 its analyses, go back to Mesquite, and choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree                 Window) Analysis &gt; Batch Architect &gt; Show Results via Instruction                 File...</font></u>. Choose the file &quot;<strong>MesquiteInstructions</strong>&quot;,                 in the same folder as the simulated matrices, and then choose                 the results file (which should be called &quot;TDTestScore.scr&quot;,).                 Mesquite should then show you a histogram of the treelength differences.               </p>              <h3>Interpreting the results</h3>              <p>An analysis of this exact sort but with 500 replicates yielded                 the following histogram:</p>              <p align="center"><img src="images/testStructure1.gif" width="370" height="409"></p>              <p>Of the 500 replicates, 439 had a value 0, 58 had a value 1, and                 3 had a value 2. You can determine these numbers by touching on                 each bar with the arrow:</p>              <p align="center"><img src="images/barValue.gif" width="152" height="145"></p>              <p>You can also see these values by going to the Text view of the                 chart (by touching the Text tab at the upper part of the chart                 window):</p>              <p align="center"><img src="images/histogramAsText.gif" width="531" height="152"></p>              <p>To see what values are in a specified percentile of the left                 or right tail of the distribution, you can choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Bar                 &amp; Line Chart) Chart> Analysis > Percentiles...</font></u>.                 In the dialog box presented, you can choose the value of the percentile,                 the color of the bar to be shown, and whether the left or right                 tails (or both) are calculated. By default, the percentile value                 is 0.05; on this example, it would be displayed as:</p>              <p align="center"><img src="images/showPercentile.gif" width="278" height="307"></p>              <p align="left">All values to the right of this red bar are thus                 in the extreme of the distribution, and our simulation would suggest                 that any value greater than or equal to 2 would occur with a frequency                 of less than 0.05. Our observation of 9 is thus an unlikely outcome                 if our hypothesis were true. Thus, we can reject our hypothesis                 at p &lt; 0.05, and conclude that trachypachids and dytiscoids                 do not form a clade. </P>              <P> As it may be hard to interpret the value of the boundaries by                 the colored vertical lines, you might want to know the exact values                 of the percentile boundaries. This information can can be gathered                 by looking at the text view for the histogram. Toward the bottom                 of the text view in the above example is the following: </p>              <p align="center"><img src="images/showPercentileText.gif" width="442" height="49"></p>              <p align="left">The percentile bar shown will be that that is closest                 to that requested, but not over it. </p>              <p align="left">The behavior of Mesquite can best be illustrated                 with another example. If you requested a percentile of 0.05, and                 if a percentile bar might be placed at 0.043 (left, below), but                 moving it one increment higher would include enough values to                 increase the percentile to 0.051 (right, below), then the percentile                 bar shown will be 0.043. </p>              <p align="center"><img src="images/boundaryRight.gif" width="218" height="193">                 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="images/boundaryLeft.gif" width="218" height="193">               </p>              <p align="left">&nbsp;</p>              <h3>Summary</h3>              <p>The hypothesis tested herein was the monophyly of a clade. The                 steps in the test are:</p>              <p>               <ul>                <li>Determine the observed value of the test statistic for the                   observed DNA sequences. In this case, the treelength of unconstrained                   most-parsimonious trees was determined using PAUP*, and was                   then subtracted from constrained (trachypachids with dytiscoids)                   treelength.</li>                <li>The best tree matching the hypothesis to be tested (trachypachids                   with dytiscoids) is inferred using the available sequence data,                   with branch lengths inferred using maximum likelihood. This                   is the model tree.</li>                <li> Values of parameters of a model of sequence evolution (gamma                   shape parameter, rate matrices, etc.) are inferred using maximum                   likelihood on the model tree for the DNA sequences.</li>                <li>The data matrix is opened in Mesquite, along with the model                   tree. Submodels and a model of character evolution are created                   within Mesquite to match those inferred.</li>                <li>Mesquite's Batch Architect is used to automate the process                   of simulating the evolution of multiple (100 or more) data matrices                   under this model, using Mesquite's Genesis package. Batch Architect                   also builds a command file for PAUP* and an instruction file                   for Mesquite so that it can interpret the results of the PAUP*                   analyses.</li>                <li>The command file is executed in PAUP*, telling PAUP* to search                   for the shortest constained and unconstrained trees for each                   of the simulated matrices, accumulating the results into a score                   file.</li>                <li>The Mesquite Instructions file is then read into Mesquite,                   and the score file is read, and Mesquite presents a histogram                   of the distribution of the test statistic (constrained treelength                   - unconstrained treelength).</li>                <li>The observed value of the test statistic is compared to the                   distribution of the statistic expected under the model as determined                   by the simulations. If the observed value is more extreme than                   that expected (say, greater than 95% of the expected values),                   the hypothesis is rejected.</li>              </ul>              <font face="Trebuchet MS"> </font></p> <font face="Trebuchet MS">               <h3>References</h3>              <p>Bell, R. T. 1966. <em>Trachypachus</em> and the origin of Hydradephaga.                 Coleopts Bull. 20:107-112.</p>              <p>Felsenstein, J. 1978. Cases in which parsimony and compatibility                 methods will be positively misleading. Systematic Zoology, 27,                                401-410.</p>              <p>Goldman, N., J. P. Anderson, and A. G. Rodrigo. 2000. Likelihood-based                 tests of topologies in phylogenetics. Syst. Biol. 49:652&#8211;670.</p>              <p>Hammond, P. M. 1979. Wing-folding mechanisms of beetles with                 special reference to investigations of Adephagan phylogeny. Pp.                 113-180 in Carabid beetles; their evolution, natural history,                 and classification (T. L. Erwin, G. E. Ball, D. R. Whitehead,                 and A. Halpern, eds.) . W. Junk, The Hague.</p>              <p>Hendy, M.D. &amp; Penny, D. 1989. A framework for the quantitative                 study of evolutionary trees. Systematic Zoology, 38, 297-309.</p>              <p>Huelsenbeck, J.P., Hillis, D.M. &amp; Jones, R. 1995. Parametric                 bootstrapping in molecular phylogenetics: Applications and performance.                                Molecular zoology: Advances, Strategies, and Protocols; Symposium                 held during Annual Meeting of the American Society of Zoologists,                                St. Louis, Missouri, USA, January 5-8, 1995 (ed. by J. D. Ferraris                 &amp; S. R. Palumbi), pp. 19-45. Wiley-Liss, Inc., New York. </p>              <p>Huelsenbeck, J.P. 1997. Is the Felsenstein zone a fly trap?                 Systematic Biology, 46, 69-74.</p>              <p>Huelsenbeck, J. P., D. M. Hillis, and R. Nielsen. 1996. A likelihood-ratio                 test of monophyly. Syst. Biol. 45:546-558.</p>              <p>Roughley, R. E. 1981. Trachypachidae and Hydradephaga (Coleoptera),                 a monophyletic unit? Pan-Pac. Entomol. 57:273-285.</p>              <p>Shull, V., A.P. Vogler, M.D. Baker, D.R. Maddison, and P.M. Hammond.                 2001. Sequence alignment of 18S ribosomal RNA and the basal relationships                 of adephagan beetles: Evidence for monophyly of aquatic families                 and the placement of Trachypachidae. Systematic Biology, 50:945-969.</p>              <p>Swofford, D. L., G. J. Olson, P. J. Waddell, and D. M. Hillis.                 1996. Phylogenetic inference. Pp. 407-514 in Molecular Systematics                 (D. M. Hillis, C. Moritz, and B. K. Mable, eds.) . Sinauer, Sunderland,                 MA.</p>              <p>Ward, R. D. 1979. Metathoracic wing structures as phylogenetic                 indicators in the Adephaga (Coleoptera). Pp. 181-191 in Carabid                 beetles; their evolution, natural history, and classification                 (T. L. Erwin, G. E. Ball, D. R. Whitehead, and A. Halpern, eds.)                 . W. Junk, The Hague.</p>              <p><HR></p>              <h4>Citation for this page</h4>              <p><font size="-1">Maddison, D.R. 2004. Testing monophyly of a group                 of beetles. Study 1 <em>in</em> Mesquite: a modular system for                 evolutionary analysis, version<!-- #BeginLibraryItem "/Library/MesquiteVersionNumber.lbi" --> 1.0<!-- #EndLibraryItem -->4,                 http://mesquiteproject.org.</font></p>              <p>&nbsp;</p>              </font> <!-- #EndEditable --></font></td>
          </tr>
        </table>
        <table border="0" cellspacing="0" cellpadding="5" width="100%">
          <tr> 
            <td align="left" valign="top"> 
              <hr>
              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite
              Home Page</font></a><font size="2"> | <a href="../../manual.html">Mesquite
              Manual</a></font></span></font></p>
              <p><font face="Trebuchet MS"><span class="smallplain">Copyright 
                &copy; 2002-2006 by <a href="http://salticidae.org/wpm/home.html">Wayne 
                P. Maddison</a> and <a href="http://david.bembidion.org">David 
                R. Maddison</a>.<br>
&nbsp;All rights reserved.</span></font></p></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</div>

<font face="Trebuchet MS"> 
</font>
</body>
<!-- #EndTemplate --></html>
