<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 -->
<head>
<!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"-->
<!-- #BeginEditable "doctitle" --> <title>Simulating DNA Evolution</title><!-- #EndEditable -->
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body bgcolor="#FFFFFF">
<div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%"> 
  <table border="0" cellspacing="0" cellpadding="0" width="100%">
    <tr> 
	  <td align="left" valign="top" height="60"> 
        <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">
          <tr> 
       		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="../images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>
            <td bgcolor="#FFFFFF"> 
              <div align="left"><font face="Trebuchet MS"><img src="../images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></font></div>
            </td>
          </tr>
		  <tr>
			<td height="42" align="left" valign="top"></td>
			<td height="42" align="left" valign="top"><font face="Trebuchet MS"><img src="../images/mesquiteLowerManual.gif"></font></td>
		  </tr>
		  <tr>
  		    <td colspan="2" height="10"></td>
		  </tr>
		  <tr>
		    <td colspan="2" height="15" align="center" valign="middle">
			<!-- Google CSE Search Box Begins  -->
				<form id="searchbox_016202157087875381635:khljhzi-ps8" action="http://www.google.com/cse">
					<input type="hidden" name="cx" value="016202157087875381635:khljhzi-ps8" />
					<input type="hidden" name="cof" value="FORID:0" />
					<input name="q" type="text" size="20" />
					<input type="submit" name="sa" value="Search Mesquite" />
				<!--<img src="http://www.google.com/coop/images/google_custom_search_smnar.gif" alt="Google Custom Search" />-->
		  		</form>
			<!-- Google CSE Search Box Ends -->
        		    </td>
		  </tr>
        </table>
      </td>
	<td>
        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">
          <tr valign="top" height="92"> 
           	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="../images/fadeRight.gif" width="72" height="160"></font></td>
			<td bgcolor="#99FF99"><img src="../images/introduction.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width = "160" bgcolor="#99FF99"> 
                <p><font face="Trebuchet MS"><span class="smallplain"><a href="../manual.html">
                  Introduction</a><br>
                  </span><a href="../whyMesquite.html">Why?</a><br>
                  <a href="../publication.html">Publication</a><br>
                  <a href="../support.html">Support</a><br>
                  <a href="../credits.html">Credits</a> </font><br>
                  <font face="Trebuchet MS"><span class="smallplain"><a href="../help.html">Help</a></span></font><br> 
                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web Site</font></a><br>
			</td>
      		<td bgcolor="#99FF99"><img src="../images/basics.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99"> 
              <p><font face="Trebuchet MS"><span class="smallplain"><a href="../help.html"></a>
                <a href="../managingFiles.html">Files</a></span> <br>
                <span class="smallplain"><a href="../menus.html">Menus</a><br>
                <a href="../windows.html">Windows</a><br>
				<a href="../charts.html">Charts</a><br>
                <a href="../scripting.html">Scripts/Macros</a> <br>
                <a href="../modules.html">Modules</a><br>
				<a href="../howMesquiteWorks.html">How</a></span></font>
              </td>
      		<td bgcolor="#99FF99"><img src="../images/objects.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">
				<p><font face="Trebuchet MS">
                <span class="smallplain"><a href="../characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="../Taxa.html"><br>
                Taxa<br>
                </a></span></font><font face="Trebuchet MS"><a href="../trees.html">Trees</a><br>
                <a href="../glossary.html">Glossary</a> </font></p>
              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="../changes.html">New</a><br>
                &nbsp;&nbsp;<a href="../changes.html">Features</a></font></em></font></p>
              </td>
      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="../images/fadeLeft.gif" width="92" height="160"></font></td>
		</tr>
    	<tr>
		    <td width="72" height="20"></td>
		<td></td>
		    <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td>&nbsp;</td>
		</tr>
		</table>
		</td>
	</tr>
  </table>
  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">
    <tr> 
      <td valign="top" align="left" width="110" height="100%"> 
        <table border="0" cellspacing="0" cellpadding="0" width="142">
          <tr> 
			<td colspan="3" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="../images/fadeDown.gif" width="100%" height="95" valign "top" border="0" alt=" "></font></p></td>
		  </tr>
          <tr> 
            <td width="6" bgcolor="#99FF99" height="35"></td>
			<td colspan="2" bgcolor="99FF99" valign="top" height="35"><p><font face="Trebuchet MS"><a href="../toc.html" style="color:#006633; font:'Trebuchet MS'; font-weight:bold; font-size:larger">Table of</a><br> 
			   &nbsp; &nbsp; <a href="../toc.html" style="color:#006633; font:'Trebuchet MS'; font-weight:bold; font-size:larger">Contents</a></font></p></td>
		  </tr>
		  <tr>
            <td width="6" bgcolor="#99FF99" height="35"></td>
            <td colspan="2" bgcolor="#99FF99" height="35"><p><font face="Trebuchet MS"><img src="../images/analyses.gif"></font></p>
              </td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>
            <td width="131" height="35" align="left" valign="middle"><p>
			  <font face="Trebuchet MS">
			  <a href="../CharacterEvolution/charEvolution.html">Character</a><br>
			    &nbsp; <a href="../CharacterEvolution/charEvolution.html">Evolution</a> <br> <br>
			  <a href="../Diversification/diversification.html">Diversification</a><br><br>
			  <a href="../molecular/molecular.html">Molecular</a><br>
			    &nbsp; <a href="../molecular/molecular.html">Data</a><br><br>
			  <a href="../popGen/popGen.html">Gene trees</a> <br>  <br>
			  <a href="RandomSimul.html">Simulations &amp;</a><br>
			    &nbsp; <a href="RandomSimul.html">Randomizations</a> <br> <br>
			  <a href="../continuous/continuous.html">Continuous</a><br>
			    &nbsp; <a href="../continuous/continuous.html">Characters</a><br> <br>
			  <a href="../otherPrograms.html">Use with</a><br>
			    &nbsp; <a href="../otherPrograms.html">Other Programs</a> <br><br>
			  <a href="../studies/index.html">Studies</a> </font></p></td>
            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr> 
            <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="5" height="92"></font></td>
            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="100%" height="92"></font></td>
          </tr>
        </table>
      </td>
      <td valign="top" width="100%" height="100%"> 
        <table width="700" border="0" cellpadding="10" cellspacing="0">
          <tr> 
            <td><font face="Trebuchet MS"><!-- #BeginEditable "body" -->               <h2>Simulating DNA sequence evolution<font face="Trebuchet MS">,                 with examples of hypothesis testing</font></h2>              <p>This section provides a brief introduction to some of the analyses                 you can do by simulating DNA sequence evolution, including:</p>              <p>               <ul>                <li><a href="#basicsimulation">Simulating DNA sequence evolution                   on a tree</a></li>                <li><a href="#complexmodels">Building more complex models</a></li>                <li><a href="#multisim">Batch Architect: Simulating, exporting,                   and analyzing multiple matrices</a></li>                <li><a href="#examples">Examples of tests of phylogenetic hypotheses</a></li>              </ul></p>              Example files for this chapter are in the &quot;Character_Simulations&quot;               folder within the &quot;examples&quot; folder within Mesquite_Folder.               <p>&nbsp;</p>              <h3><a name="basicsimulation"></a>Simulating DNA sequence evolution                 on a tree</h3>              <p>First, let's try evolving DNA sequences up the branches of a                 model tree, thereby creating an entire simulated matrix. Open                 the example file &quot;<strong>01-modelTree.nex</strong>&quot;                 and you will be presented with a tree. For the moment it doesn't                 matter where this tree came from; the important thing to note                 is that it has branch lengths inferred from a matrix of 18S rDNA;                 this matrix is also included in the file.</p>              <p>In order to simulate DNA sequences, you need:</p>              <ul>                <li> A tree, with branch lengths specified. A tree with branch                   lengths is included in this example file.</li>                <li>A specification of how the DNA sequences evolve. This model                   could have many elements. One common form of model specifies                   the relative frequencies of A, C, G, and T; the relative rates                   of change of the different characters; and the relative rates                   of change between A and C, A and G, and so on.</li>              </ul>              <h4>Creating a model of character evolution</h4>              <p>For sake of simplicity, let's presume that A, C, G, and T occur                 at equal frequencies, and that all pairs of nucleotides have equal                 rates of change between them. To make it interesting, though,                 let's presume that some characters evolve more quickly than others,                 with the distribution of rates of change following a discrete                 gamma distribution with shape parameter 1.5. If one presumes (as                 we will) that there were four categories of rates, then this model                 presumes that one quarter of the characters have rate 0.225, one                 quarter 0.589, one quarter 1.05, and the rest 2.136, as shown                 in the following figure:</p>              <p align="center"><img src="images/gamma15.gif" width="233" height="51"></p>              <p>To create this model of character rate variation in Mesquite,                 choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree)                 Characters&gt;New Character Submodel&gt;Gamma Rate Variation Model...</font></u>                 You will be presented with a dialog box in which you can choose                 the name of the model you will create; you might call it &quot;my                 gamma model&quot;. You will then be presented with a dialog box                 allowing you to specify the nature of the model; set the shape                 parameter to 1.5:</p>              <p align="center"><img src="images/gamma15dlog.gif" width="218" height="159"></p>              <p>Now we need to create a composite model of character evolution                 that specifies both this gamma rate variation model, the state                 frequency model, and the state-to-state rate model. To do this,                 choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree)                 Characters&gt;New Character Model&gt;Composite DNA Simulation                 Model...</font></u>, name the model (perhaps call it &quot;my                 full model&quot;), and you will be presented with a dialog box                 in which you can choose the various submodels:</p>              <p align="center"><img src="images/fullmodel.gif" width="373" height="323"></p>              <p align="left">The only element of this model we will change is                 the character rates model; choose &quot;my gamma model&quot; from                 the drop-down menu to select the gamma distribution model you                 created as the model of site-to-site rate variation. The model                 &quot;my full model&quot; is now ready to use. </p>              <p align="left">Note that the dialog box in which you edited &quot;my                 full model&quot; has button with a ? on it (<img src="../images/helpButton.gif" width="29" height="28" align="absmiddle">).                 When a button like this is present in a Mesquite window, touching                 on it will display a small note with information about how to                 use the window's feature.</p>              <h4 align="left">Simulating evolution</h4>              <p align="left">Now choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree)                 Characters&gt;Make New Matrix from&gt;Simulated Matrices on Current                 Tree</font></u>. This will ask Mesquite to simulate evolution                 of characters up the current tree in the tree window to produce                 a new matrix. You will be asked what sort of character simulator                 to use. As we wish to evolve a DNA sequence up the tree's branches,                 choose &quot;Evolve DNA characters&quot; from the list. (There                 might be other options presented, such as Evolve Continuous Characters,                 but we don't wish to try that now.) You will then need to chose                 the model of DNA sequence evolution; choose the composite one                 you created earlier called &quot;full model&quot;. When it queries                 for the number of character, choose a large number, such as 2000.                 When it asks for the name of the matrix to be created, call it                 &quot;simulated&quot; (although you could choose another name,                 if you wish). Mesquite will then simulate evolution, and a matrix                 will be produced, such as this one:</p>              <p align="center"><img src="images/simMatrix.gif" width="364" height="256"></p>              <h4 align="left">Examining the results of the simulation</h4>              <p align="left">Let's see whether the rate variation between characters                 suggested by this simulated matrix matches a gamma distribution                 with shape parameter 1.5, as used in the model. First, save the                 matrix to a file so that you can read it into PAUP*. Choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                 Matrix) Characters&gt;Save Copy of Matrix&gt;simulated</font></u>                 to save a copy of your newly created matrix; perhaps name the                 file &quot;Simulated Matrix&quot;.</p>              <p align="left">In PAUP* 4, execute the file Simulated Matrix, and                 then execute the following commands:</p>              <pre align="left"><font face="Courier New, Courier, mono">      nj;        lset nst=1 basefreq=equal rates=gamma shape=estimate;      lscore 1;</font></pre>              <p align="left">(If you don't know how to give PAUP* commands, please                 read your PAUP* documentation. There are equivalent ways to give                 these commands in the MacOS versions using dialog boxes, but it                 is much easier for us to describe what to do with these text commands.)               </p>              <p align="left">The first command (&quot;nj;&quot;) will find a                 neighbor-joining tree. While this might not be the best way to                 get a tree, but it will be fast, and give a tree that is good                 enough for our purposes. The second command (&quot;lset...&quot;)                 sets the model used for likelihood calculations to be the model                 used by Mesquite in the simulations, except that the shape parameter                 used is not specified, but is instead estimated from the data.                 The third command (&quot;lscore 1;&quot;) tells PAUP* to calculate                 the likelihood of the tree, in the course of which it will also                 estimate the shape parameter of the gamma distribution using likelhood.                 You might end up with PAUP* reporting something like this:</p>              <pre align="left"><font face="Courier New, Courier, mono">       Tree 1       -------------------       -ln L 19742.05025       Shape 1.583552 </font></pre>              <p align="left">The shape parameter should be around 1.5.</p>              <h3 align="left"><br>                <a name="complexmodels"></a> Building more complex models</h3>              <p align="left">More complex composite models of DNA evolution can                 be built by choosing among the following:</p>              <p align="left"><em>Models of root state frequencies: </em>               <ul>                <li>Equal Frequencies: equal frequencies of states</li>                <li>Empirical Frequencies: frequencies of states the same as that                   found in an existing character matrix</li>                <li>User-specified Nucleotide Frequencies: frequencies of nucleotides                   specified by the user. You can create a model of this sort by                   choosing <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                   Matrix) Characters&gt;New Character Submodel&gt;User-specified                   Nucleotide Frequencies...</font></u> and then entering the appropriate                   values in the dialog box.</li>              </ul></p>              <p align="left"><em>Models of equilibrium state frequencies on other                 branches:</em>               <ul>                <li>Equal Frequencies: equal frequences of states</li>                <li>Empirical Frequencies: frequencies of states the same as that                   found in an existing character matrix</li>                <li>User-specified Nucleotide Frequencies: frequencies of nucleotides                   specified by the user. You can create a model of this sort by                   choosing <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                   Matrix) Characters&gt;New Character Submodel&gt;User-specified                   Nucleotide Frequencies...</font></u> and then entering the appropriate                   values in the dialog box.</li>              </ul></p>              <p align="left">The above two submodels corresponds in PAUP* to                 the specifying the base frequencies in the likelihood settings;                 note that PAUP* does not allow separate specification of the model                 of frequences and the root and at other branches.</p>              <p align="left"><em>Models of rate variation among characters:</em>               <ul>                <li>Equal Rates: all characters evolve at the same rate.</li>                <li>Codon Position Rates Model: a model specifying the relative                   rates of the different codon positions. If you wish to have                   the codon positions match those in an existing matrix, then                   codon positions need to be specified for a DNA matrix. To do                   this, choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                   Matrix) Characters&gt;List of Characters</font></u> to see the                   current codon positions. Select the characters in this list                   window, and then touch on the title of the column &quot;Codon                   Position&quot;. A drop-down menu will appear in which you can                   choose to set the codon positions to a sequence like 123123123...                   as appropriate.</li>                <li>Gamma Rates Model: a model specifying that rates of characters                   evolve according to a gamma distribution.</li>                <li>Gamma Invar Rates Model: a model specifying that a proportion                   of the characters are invariant, and the remainder follow a                   gamma distribution.</li>                <li>Proportion Invariant Model: a model specifying that a proportion                   of the characters are invariant, and the remainder evolve at                   one rate.</li>              </ul></p>              <p align="left">The first model (&quot;Equal Rates&quot;) is built                 in; you can create a model of the other kinds by choosing the                 appropriate items from the <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                 Matrix) Characters&gt;New Character Submodel</font></u> menu and                 then entering the values in the dialog box.</p>              <p align="left"><em>Model of rate matrices</em>:               <ul>                <li>Single Rate: all changes between nucleotides occur at the                   same rate.</li>                <li>Ti/Tv Rate Matrix Model: the two-parameter rate matrix model                   in which transversions occur at a different rate than transversions.</li>                <li>GTR Rate Matrix Model: the General Time Reversible, six-parameter                   rate matrix model in which you can specify rates of each type                   of nucleotide change. </li>              </ul></p>              <p align="left">The first model (&quot;Single Rate&quot;) is built                 in; you can create a model of the other kinds by choosing the                 appropriate items from the <u><font size="-1" face="Arial, Helvetica, sans-serif">(Character                 Matrix) Characters&gt;New Character Submodel</font></u> menu and                 then entering the values in the dialog box.</p>              <h3 align="left"><a name="multisim"></a>Batch Architect: Simulating,                 exporting, and analyzing multiple matrices</h3>              <p align="left">Mesquite's <strong>Batch Architect</strong> package                 can be used to create multiple simulated matrices, export them                 to files, and create one or more files that provide instructions                 to programs to analyze the files produced. These instruction files                 are called &quot;batch files&quot;. </p>              <p align="left">One could, for example, examine the quality of Mesquite's                 simulation algorithms by simulating 100 matrices using a gamma                 model of character rate variation. In addition to creating 100                 files with matrices, Mesquite would create a batch file containing                 instructions for PAUP* to estimate the gamma shape parameter for                 each of the matrices and save the resulting values to a score                 file. Mesquite would also create an file that would provide instructions                 to Mesquite to allow it to read PAUP*'s score file and display                 the resulting distribution of estimated gamma shape parameters                 in a chart. As a concrete example, simulation of 1000 matrices                 of 2000 characters each under a model with a gamma shape parameter                 of 2.0, estimating of the shape parameter by PAUP* for each of                 these matrices, and processing of the results by Mesquite yielded                 the following histogram:</p>              <p align="center"><img src="images/gammaTest.gif" width="422" height="341"></p>              <p align="left">In this particular analysis, most matrices were                 estimated to have a gamma shape parameter close to that used in                 the model that generated the data; the average estimated gamma                 shape parameter over the 1000 replicates was 2.017.</p>              <h4 align="left">Conducting multiple simulations</h4>              <p align="left">To conduct an analysis similar to this one, open                 up the example file &quot;<strong>02-gammaTest.nex</strong>&quot;.                 A complete model of DNA sequence evolution has already been added                 to that file, under the name &quot;gamma2model&quot;. It specifies,                 among other components, character rate variation following a gamma                 distribution with shape parameter 2.0. First, you will need to                 ask Mesquite to evolve several matrices up the branches of the                 phylogeny. Choose <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree                 Window) Analysis &gt; Matrices &amp; Batch Files &gt; Export Matrices                 &amp; Batch Files...</font></u>. You will be presented with a                 choice of the source of matrices; choose &quot;Simulated Matrices                 on Current Tree&quot;, and press OK; for the character simulator,                 choose &quot;Evolve DNA characters&quot;. When asked to choose                 the model, choose &quot;gamma2model&quot;. </p>              <p align="left">You will now be presented with the main dialog box                 that you will need to master to use Mesquite's simulation tools.                 It looks approximately like this:</p>              <p align="center"><img src="images/exportMatrixBatch.gif" width="336" height="356"></p>              <p align="left">The top part of the dialog box allows one to choose                 the base part of the name of the simulated matrices that will                 be produced, as well as to specify how many matrices will be produced                 (the &quot;number of replicates&quot;). Let's call have the file                 names begin with &quot;gammaTest&quot;, so enter that for the                 base name. In the interests of time, you can leave the number                 of replicates be only 10 (a more thorough analysis would involve                 many more replicates).</p>              <p align="left">In addition to creating the matrices, Mesquite will                 also produce one or two files accompanying those matrices that                 specify commands to be used by various software (e.g. PAUP, NONA,                 even Mesquite) to analyze those matrices. The manner in which                 these accessory files or &quot;batch files&quot; are written is                 controlled by the lower portion of the dialog box. The content                 of the batch files is specified by a &quot;batch file template&quot;.                 There are some templates supplied by Mesquite; you can also build                 your own. The template we need to use for this example is not                 built into Mesquite; you will need to load it in from a file.                 To do this, go into the Templates Manager, by pressing the Edit                 Templates button. In the dialog that is presented, you can see                 the build-in templates:</p>              <p align="center"><img src="images/templateManager.gif" width="275" height="242"></p>              <p align="left">To load in the template we need to use, press the                 Load button, and open the &quot;<strong>GammaTest.template</strong>&quot;                 file that is in the &quot;templates&quot; folder within the Simulations                 examples folder. The template Gamma Model Test will now be added                 to the templates list. (You can see the nature of the template                 by selecting it in this list and pressing &quot;Edit&quot;, but                 you needn't do this now.)</p>              <p align="left">Once the template is loaded, press Done to return                 to the Export Matrices &amp; Batch Files dialog box. Make sure                 &quot;Gamma Model Test&quot; is selected, and press &quot;OK&quot;.                 You will now be asked to choose a folder to store the created                 files. As multiple files will be created, you may wish to create                 a new, empty folder to house the files. Once the location is chosen,                 you will be asked (after perhaps dismissing a notice or two) for                 the number of characters to be simulated within each matrix. Choose                 a fairly high number, such as 2000. Mesquite will now proceed                 to create through simulation all 10 files, plus two batch files.               </p>              <p align="left">One of these batch files will be a file containing                 PAUP commands that will instruct PAUP* to read in each matrix                 file, quickly build a preliminary tree (using neighbor-joining,                 which should be good enough for our purposes), and estimate the                 gamma shape parameter using likelihood. The resulting value will                 be saved to a PAUP &quot;scorefile&quot;. Open up PAUP* 4 (make                 sure you have the latest version of PAUP*), and ask it to execute                 the file &quot;<strong>gammaCommands.nex</strong>&quot;. After                 PAUP has finished its analysis, go back to Mesquite, and choose                 <u><font size="-1" face="Arial, Helvetica, sans-serif">(Tree Window)                 Analysis &gt; Batch Architect&gt; Show Results via Instruction                 File...</font></u>. You will be asked to first choose an &quot;instruction                 file&quot;, which is a text file telling Mesquite how to interpret                 another file. Choose the file &quot;<strong>MesquiteInstructions</strong>&quot;,                 which is the second batch file created by the Gamma Model Test                 template. After Mesquite processes the instruction file, it will                 ask you to find the results file, which in this case is the score                 file created by PAUP*. It should be called &quot;<strong>gammaTestScore.scr</strong>&quot;                 (if the base name of the matrices was &quot;gammaTest&quot;);                 open it when asked for the results file. You will then be presented                 with a histogram of the estimated gamma shape parameter values,                 that will look approximately like this:</p>              <p align="center"><img src="images/gamma10reps.gif" width="343" height="278"></p>              <p align="left">Of course, your exact values should differ from                 these, but should cluster around 2.0. If you repeated this, but                 with many more replicates, then you should see a bell-shaped curve,                 as shown above.</p>              <h4 align="left">Testing simulations of more complex models</h4>              <p align="left">A test of accuracy of Mesquite's more complex models                 can also be done. For example, there are seven parameters in model                 that includes a General Time Reversible model of DNA evolution                 and assuming a proportion of the characters are invariant with                 the remainder following a gamma distribution. In 1000 replicates                 of 2000 characters evolved up the branches of a 49-taxon tree,                 the average value of these parameters as estimated by PAUP* is                 close (within 1.3 %) of the value used in model used in Mesquite,                 as shown by the results of one particular analysis:</p>              <table width="75%" border="0" align="center" cellpadding="4" cellspacing="2">                <tr>                   <td width="36%"><strong>Param<font face="Trebuchet MS">eter</font></strong></td>                  <td width="33%"><strong><font face="Trebuchet MS">Value in model</font></strong></td>                  <td width="31%"><strong>Estimated value</strong></td>                  <td width="31%"><strong><font face="Trebuchet MS">Difference</font></strong></td>                </tr>                <tr>                   <td>Rate(A&lt;-&gt;C)</td>                  <td>1.8700052</td>                  <td>1.87<font face="Trebuchet MS">771</font></td>                  <td><font face="Trebuchet MS">0.41%</font></td>                </tr>                <tr>                   <td>Rate(A&lt;-&gt;<font face="Trebuchet MS">G</font>)</td>                  <td>4.2537253</td>                  <td>4.2<font face="Trebuchet MS">6319</font></td>                  <td><font face="Trebuchet MS">0.22%</font></td>                </tr>                <tr>                   <td>Rate(A&lt;-&gt;<font face="Trebuchet MS">T</font>)</td>                  <td>2.5286454</td>                  <td>2.5<font face="Trebuchet MS">3006</font></td>                  <td><font face="Trebuchet MS">0.06%</font></td>                </tr>                <tr>                   <td>Rate(<font face="Trebuchet MS">C</font>&lt;-&gt;<font face="Trebuchet MS">G</font>)</td>                  <td>0.626305</td>                  <td>0.6<font face="Trebuchet MS">315</font>2</td>                  <td><font face="Trebuchet MS">0.83%</font></td>                </tr>                <tr>                   <td>Rate(<font face="Trebuchet MS">C</font>&lt;-&gt;<font face="Trebuchet MS">T</font>)</td>                  <td>8.735118</td>                  <td>8.<font face="Trebuchet MS">77468</font></td>                  <td><font face="Trebuchet MS">0.45%</font></td>                </tr>                <tr>                   <td>propo<font face="Trebuchet MS">rtion invariant</font></td>                  <td>0.506903</td>                  <td>0.509<font face="Trebuchet MS">03</font></td>                  <td><font face="Trebuchet MS">0.42%</font></td>                </tr>                <tr>                   <td>gamma <br>                    shape</td>                  <td>0.438522</td>                  <td>0.444<font face="Trebuchet MS">22</font></td>                  <td><font face="Trebuchet MS">1.30%</font></td>                </tr>              </table>              <p align="left">&nbsp;</p>              <h3 align="left"><a name="examples" id="examples"></a>Examples of                 tests of phylogenetic hypotheses</h3>              <p align="left">You can use Mesquite's Batch Architect and Genesis                 packages to build your own statistical tests of various phylogenetic                 hypotheses. Some example tests are presented in the following                 pages:</p>              <ul>                <li><a href="../studies/study001/index.html">Testing monophyly                   of a group of beetles</a> </li>                <li><a href="../studies/study002/index.html">Are strepsipterans                   related to flies? Exploring long branch attraction</a></li>              </ul>              <p align="center">&nbsp;</p>              <h3><font face="Trebuchet MS">Other analyses that Mesquite can do</font></h3>              <p><font face="Trebuchet MS"> The examples given about used the               Genesis package in Mesquite to simulate the evolution of characters               under a specified model of DNA evolution, and up the branches of               the current tree on the screen. Many more sorts of analyses can               be done, using different sources of characters other than simulated               DNA data, different sources of trees other than the current tree               on the screen, and different calculations by other programs.               <p align="left"><em>Some of the different sources of character matrices                 include: </em>               <ul>                <li>Simulators that evolve data other than DNA data, including                   continuously valued data.</li>                <li>Random modifications of existing matrices, including by non-parametric                   resampling (as used in classic bootstrap methods), random reshuffling                   of data, and jackknifing.</li>              </ul></p>              <p align="left"><em>Some of the different sources of trees include:                 </em>               <ul>                <li>Simulations using markovian models of speciation</li>                <li>Coalescence simulations of gene trees</li>                <li>Modifications of an existing tree by random modification of                   branch lengths</li>                <li>Modifications of an existing tree by randomly pruning a fraction                   of the taxa or by randomly adding taxa</li>                <li>Randomly sampled trees among all possible trees</li>                <li>Trees generated by randomly reshuffling taxa at their tips</li>              </ul></p>              <p>&nbsp;</p>              <h3>References</h3>              <p>Shull, V., A.P. Vogler, M.D. Baker, D.R. Maddison, and P.M. Hammond.                 2001. Sequence alignment of 18S ribosomal RNA and the basal relationships                 of adephagan beetles: Evidence for monophyly of aquatic families                 and the placement of Trachypachidae. Systematic Biology, 50:945-969.</p>              <p>&nbsp;</p>              <p>&nbsp;</p>              </font> <!-- #EndEditable --></font></td>
          </tr>
        </table>
        <table border="0" cellspacing="0" cellpadding="5" width="100%">
          <tr> 
            <td align="left" valign="top"> 
              <hr>
              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite
              Home Page</font></a><font size="2"> | <a href="../manual.html">Mesquite
              Manual</a></font></span></font></p>
              <p><font face="Trebuchet MS"><span class="smallplain">Copyright 
                &copy; 2002-2007 by <a href="http://salticidae.org/wpm/home.html">Wayne 
                P. Maddison</a> and <a href="http://david.bembidion.org">David 
                R. Maddison</a>.<br>
&nbsp;All rights reserved.</span></font></p></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</div>

<font face="Trebuchet MS"> 
</font>
</body>
<!-- #EndTemplate --></html>
