<!--This file created 1.4.99 9:31 PM by Claris Home Page version 2.0-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 -->
<head>
<!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"-->
<!-- #BeginEditable "doctitle" -->
<title>Module index</title>
<!-- #EndEditable -->
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body bgcolor="#FFFFFF">
<div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%"> 
  <table border="0" cellspacing="0" cellpadding="0" width="100%">
    <tr> 
	  <td align="left" valign="top" height="60"> 
        <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">
          <tr> 
       		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>
            <td bgcolor="#FFFFFF"> 
              <div align="left"><font face="Trebuchet MS"><a href="manual.html"><img src="images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></a></font></div>
            </td>
          </tr>
		  <tr>
			<td height="42" align="left" valign="top"></td>
			<td height="42" align="left" valign="top"><font face="Trebuchet MS"><a href="manual.html"><img src="images/mesquiteLowerManual.gif" border="0"></a></font></td>
		  </tr>
<!--	  <tr>
  		    <td colspan="2" height="10"></td>
		  </tr>
	      <tr>
		    <td colspan="2" height="15" align="center" valign="middle">
			<!-- Google CSE Search Box Begins
				<form id="searchbox_016202157087875381635:khljhzi-ps8" action="http://www.google.com/cse">
					<input type="hidden" name="cx" value="016202157087875381635:khljhzi-ps8" />
					<input type="hidden" name="cof" value="FORID:0" />
					<input name="q" type="text" size="20" />
					<input type="submit" name="sa" value="Search Mesquite" />
				<!--<img src="http://www.google.com/coop/images/google_custom_search_smnar.gif" alt="Google Custom Search" />
		  		</form>
			<!-- Google CSE Search Box Ends
        		    </td>
		  </tr> -->
        </table>
      </td>
	<td>
        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">
          <tr valign="top" height="92"> 
           	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="images/fadeRight.gif" width="72" height="160"></font></td>
			<td bgcolor="#99FF99"><img src="images/introduction.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width = "160" bgcolor="#99FF99"> 
                <p><font face="Trebuchet MS"><span class="smallplain"><a href="manual.html">
                  Introduction</a><br>
                  </span><a href="whyMesquite.html">Why?</a><br>
                  <a href="publication.html#howToCite">How to Cite</a>                <br>
                  <a href="publication.html">Publication</a><br>
                  <a href="support.html">Support</a><br>
                  <a href="credits.html">Credits</a> </font><br>
                  <font face="Trebuchet MS"><span class="smallplain"><a href="help.html">Help</a></span></font><br>
                  <a href="Help/MesquiteFAQ.html"><font face="Trebuchet MS">FAQ</font></a><br> 
                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web Site</font></a><br>
			</td>
      		<td bgcolor="#99FF99"><img src="images/basics.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99"> 
              <p><font face="Trebuchet MS"><span class="smallplain"><a href="help.html"></a>
                <a href="simplification.html">Simplicity</a><br>
                <a href="managingFiles.html">Files</a></span> <br>
                <span class="smallplain"><a href="menus.html">Menus</a><br>
                <a href="windows.html">Windows</a><br>
				<a href="charts.html">Charts</a><br>
                <a href="scripting.html">Scripts/Macros</a> <br>
                <a href="modules.html">Modules</a><br>
				<a href="howMesquiteWorks.html">How</a></span></font>
            </td>
      		<td bgcolor="#99FF99"><img src="images/objects.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">
				<p><font face="Trebuchet MS">
                <span class="smallplain"><a href="characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="Taxa.html"><br>
                Taxa<br>
                </a></span></font><font face="Trebuchet MS"><a href="trees.html">Trees</a><br>
                <a href="glossary.html">Glossary</a> </font></p>
              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="changes.html">New</a><br>
                &nbsp;&nbsp;<a href="changes.html">Features</a></font></em></font></p>
              </td>
      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="images/fadeLeft.gif" width="92" height="160"></font></td>
		</tr>
    	<tr>
		    <td width="72" height="20"></td>
		<td></td>
		    <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td>&nbsp;</td>
		</tr>
		</table>
		</td>
	</tr>
  </table>
  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">
    <tr> 
      <td valign="top" align="left" width="110" height="100%"> 
        <table border="0" cellspacing="0" cellpadding="0" width="142">
          <tr> 
			<td colspan="3" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="images/fadeDown.gif" width="100%" height="95" valign "top" border="0" alt=" "></font></p></td>
		  </tr>
          <tr> 
            <td width="6" bgcolor="#99FF99" height="35"></td>
			<td colspan="2" bgcolor="99FF99" valign="top" height="35"><p><font face="Trebuchet MS"><a href="toc.html">Table of</a><br> 
			   &nbsp; &nbsp; <a href="toc.html">Contents</a></font></p></td>
		  </tr>
		  <tr>
            <td width="6" bgcolor="#99FF99" height="35"></td>
            <td colspan="2" bgcolor="#99FF99" height="35"><p><font face="Trebuchet MS"><img src="images/analyses.gif"></font></p>
              </td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>
            <td width="131" height="35" align="left" valign="middle"><p>
			  <font face="Trebuchet MS">
			  <a href="CharacterEvolution/charEvolution.html">Character</a><br>
			    &nbsp; <a href="CharacterEvolution/charEvolution.html">Evolution</a> <br> <br>
			  <a href="Diversification/diversification.html">Diversification</a><br><br>
			  <a href="molecular/molecular.html">Molecular</a><br>
			    &nbsp; <a href="molecular/molecular.html">Data</a><br><br>
			  <a href="popGen/popGen.html">Gene trees</a> <br>  <br>
			  <a href="RandomSimul/RandomSimul.html">Simulations &amp;</a><br>
			    &nbsp; <a href="RandomSimul/RandomSimul.html">Randomizations</a> <br> <br>
			  <a href="continuous/continuous.html">Continuous</a><br>
			    &nbsp; <a href="continuous/continuous.html">Characters</a><br> <br>
			  <a href="otherPrograms.html">Use with</a><br>
			    &nbsp; <a href="otherPrograms.html">Other Programs</a> <br><br>
			  <a href="studies/index.html">Studies</a> </font></p></td>
            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr> 
            <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="images/fadeUp.gif" width="5" height="92"></font></td>
            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="images/fadeUp.gif" width="100%" height="92"></font></td>
          </tr>
        </table>
      </td>
      <td valign="top" width="100%" height="100%"> 
        <table width="700" border="0" cellpadding="10" cellspacing="0">
          <tr> 
            <td><font face="Trebuchet MS"><!-- #BeginEditable "body" --> 
              <h2>Mesquite's standard modules</h2>
              <p>The following is a list of the modules installed by default in 
                this version of Mesquite, along with the explanations of what 
                they do.</p>
              <ul>
<li>Absolute Distance (General)  &#8212; Absolute distance from a character matrix.</li>
<li>ACGT Compositional Bias  &#8212; Returns continuous characters which are the proportion G+C, and separately A, C, G, T, for each taxon in a DNA data set.</li>
<li>Add item...  &#8212; Adds an item in each cell of a continuous matrix.</li>
<li>Add Noise (Gaussian)  &#8212; Adds normally distributed noise to existing states.  Only the first item of a continuous matrix is modified.</li>
<li>Add Noise to Branch Lengths  &#8212; Adds noise to branch lengths of tree.  Noise is Normally distributed, with variance as selected.  By default the variance is be proportional to current branch length, so that if branch length is 10 and you've indicated a variance multiplier of 0.1, the noise added will have a variance of 1.0.  Negative branch lengths are not allowed, and are changed to zero.</li>
<li>Add Note On Tree  &#8212; Draws editable notes in a tree display.</li>
<li>Add Taxa To Tree  &#8212; To a tree with some taxa excluded, adds taxa.</li>
<li>Add text to taxon names  &#8212; Adds text to taxon names.</li>
<li>Add value  &#8212; Alters continuous data by adding a value.  All items of the matrix are similarly modified.</li>
<li>Add/Delete Data  &#8212; Adds and deletes characters and taxa in a Character Matrix editor.  This can be done through the Add characters..., Add taxa..., and Delete Selected menu items and via the tools that add taxa and characters where touched.</li>
<li>Adjust scaling factor of lineage widths  &#8212; Provides a tool with which to adjust the scaling factor used for lineage widths in a tree window.</li>
<li>Align DNA to Protein  &#8212; Realigns a DNA matrix to match the alignment in an amino acid alignment.</li>
<li>Align Package  &#8212; Align is a Mesquite package providing tools for the alignment of sequence data.</li>
<li>Align Score/RC Align Score To Ref.  &#8212; Reports the Alignment Score / Reverse Complement Alignment Score for a taxon in a data matrix. The score for each taxon is the cost of aligning that taxon's sequence against the reference taxon using Mesquite's default pairwise aligner, divided by the score for aligning the reverse complement of the sequence against the reference.  Scores  greater than one mean that alignment is better if one of the sequences is reverse complemented .</li>
<li>Align Sequences  &#8212; Sends the selected sequence to be aligned.</li>
<li>Align To Dropped  &#8212; Supplies an alignment tool that can be used on a set of sequences.  Sequences dropped by this tool on another sequence will be aligned to that other sequence (pairwise).</li>
<li>Alignment Score To Ref.  &#8212; Reports the Alignment Score for a taxon in a data matrix. The score for each taxon is the cost of aligning that taxon's sequence against the reference taxon using Mesquite's default pairwise aligner.  Lower scores mean a better alignment.</li>
<li>All Branch Lengths to 1.0  &#8212; Assigns a value of 1.0 for branch length for all of a tree's branches.</li>
<li>All Rerootings  &#8212; Reroots tree at all nodes.</li>
<li>All Trees  &#8212; Supplies all possible dichotomous rooted or unrooted trees (only works if number of taxa is <=11).</li>
<li>All Unassigned Branch Lengths to 1.0  &#8212; Assigns a value of 1.0 for branch length for all of a tree's branches that have not yet had a branch length assigned.</li>
<li>Alter Branch Lengths  &#8212; Alters branch lengths of selected trees.</li>
<li>Alter Data  &#8212; Manages data-transforming modules.</li>
<li>Alter Trees  &#8212; Alters selected trees.</li>
<li>Alternative Names  &#8212; Lists the alternative names for the taxon.</li>
<li>Amino Acid Molecular Weight  &#8212; Returns molecular weight of amino acid</li>
<li>Amino Acid Properties  &#8212; Returns property of amino acid</li>
<li>Among-group PCA  &#8212; Performs among-group principal components analysis on a continous-valued matrix.</li>
<li>Ancestral States All  &#8212; Reconstructs ancestral states at nodes for all characters supplied by character source.</li>
<li>Ancestral States Reconstruction Package Introduction  &#8212; The ancestral states reconstruction package provides modules to coordinate and display reconstruction of ancestral states on the branches of trees.</li>
<li>Annotate Branches  &#8212; Supplies a tool for tree windows to attach and view footnotes for branches.</li>
<li>Annotation Panel module  &#8212; Provides tools with which to attach notes (including images) to cells of the matrix and show them.</li>
<li>Annotation Panel module (list of characters)  &#8212; Provides tools with which to attach notes (including images) to characters and show them.</li>
<li>Annotation Panel module (list of taaa)  &#8212; Provides tools with which to attach notes (including images) to characters and show them.</li>
<li>Append numbers  &#8212; Appends to each taxon name the number of the taxon. This will ensure that each taxon name is unique.</li>
<li>Arbitrarily Resolve State Ambiguities  &#8212; Alters categorical data by arbitrarily resolving ambiguous sites to one of the possible states.</li>
<li>Arbitrarily Ultrametricize  &#8212; Adjusts a tree's branch lengths so that the distances among terminal taxa are ultrametric (i.e. like a molecular clock, all tips reaching to same level).  This is not done with any sophisticated smoothing algorithm; rather, branches are just stretched until they reach to same level.</li>
<li>Assign All Branch Lengths  &#8212; Assigns a value for branch length for all of a tree's branches.</li>
<li>Assign Branch Lengths From Node Names  &#8212; Assigns a value for branch length for all of a tree's branches based upon node names that represent numbers.</li>
<li>Assign Branch Lengths Less than or Equal to Zero...  &#8212; Assigns a user-provided value to all branches with lengths less than or equal to zero.</li>
<li>Assign Selected Branch Lengths  &#8212; Assigns a value for branch length for all of a tree's selected branches.</li>
<li>Assign taxon names  &#8212; Assigns to each taxon name a string followed by a number.  For example, if the string is "g", then taxon 3 will be assigned the name g3.</li>
<li>Assigned Colors  &#8212; Provides a tool with which to color cells of a matrix.</li>
<li>Associated Taxa  &#8212; Lists and edits what other taxa (e.g. contained or containing) are associated with these.</li>
<li>Asymmetrical 2-param. Markov-k Model  &#8212; Defines and maintains simple Markov k-state asymmetrical 2-parameter stochastic models of character evolution.</li>
<li>Asymmetry Likelihood Ratio Test  &#8212; Calculates the test statistic for the likelihood ratio test comparing the asymmetrical and one parameter models [2ln(L(Asymm.)/L(Mk1)], on a tree for a given character.</li>
<li>Attached Value  &#8212; Supplies a number attached to the tree</li>
<li>Attachment  &#8212; Displays an item attached to trees.</li>
<li>Augment Tree Randomly  &#8212; Augments tree by random placement of excluded taxa.  There are three modes: adding only to original branches, ignoring lengths; adding to any branch, considering branch length; adding to any branch, ignoring branch length.  With lengths considered, probability of placing taxon on a branch is proportional to the branch's length.</li>
<li>Average of Several Distances  &#8212; Average of several distances among taxa, e.g. average from several data matrices.</li>
<li>Average Taxon Value among Trees  &#8212; Averages among trees a value calculated for a taxon using a tree.</li>
<li>Average Tree Depth  &#8212; Calculates the average path length (in branch length) from terminals to the root, treating unassigned lengths as 0.  The length of the root is not counted.</li>
<li>Average Tree Value  &#8212; Calculates the average of some value for trees in a tree block.</li>
<li>Balls &amp; Sticks  &#8212; Draws trees with spots and the nodes and thin lines for branches.</li>
<li>Bar &amp; Line Chart  &#8212; Helps make bar &amp; line charts.</li>
<li>Bar &amp; Line Chart for Character Matrices  &#8212; Shows chart displaying values (such as treelength, etc.) for a series of character matrices.</li>
<li>Bar &amp; Line Chart for Characters  &#8212; Shows chart displaying values (such as likelihoods, parameter values, parsimony counts, etc.) for a series of characters.</li>
<li>Bar &amp; Line Chart for Taxa  &#8212; Makes a chart displaying values (such as proportion of gaps in a sequence, etc.) for each of a series of taxa.</li>
<li>Bar &amp; Line Chart for Trees  &#8212; Makes a chart displaying values (such as likelihoods, parsimony scores, imbalance statistics, correlations,etc.) for a series of trees.</li>
<li>Basic Draw Names for Tree Display  &#8212; Draws taxon names on a tree.  Chooses orientation of names according to orientation of tree.</li>
<li>Basic File Coordinator  &#8212; Coordinates the reading, maintenance and linkages of files, which include Taxa, Trees, CharacterData, and other objects. This module is hired directly by the Mesquite trunk module.</li>
<li>Basic Tree Draw Coordinator  &#8212; Coordinates the drawing of a tree by maintaining the basic TreeDisplay and by hiring a DrawTree module.</li>
<li>Basic Utilities Package Introduction  &#8212; Provides basic utilities for Mesquite.</li>
<li>Batch Architect Package Introduction  &#8212; Provides facilities to coordinate the simulation of data and generate batch files for replicated analyses.</li>
<li>Batch File Template Manager  &#8212; Manages templates used in saving batch files.</li>
<li>Birth/Death Process Trees  &#8212; Generates tree by simple birth/death model with a constant rate of speciation (birth) and of extinction (death).</li>
<li>BiSSE Ln Likelihood Difference  &#8212; Calculates the difference in log likelihoods between two BiSSE speciation/extinction models.</li>
<li>BiSSE Net Diversification Likelihood  &#8212; Calculates likelihood with a tree of a species diversification model whose rates (r = spec. - ext.; a = spec./ext.) depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</li>
<li>BiSSE Net Diversification Likelihood (Calculator)  &#8212; Calculates likelihood with a tree of a species diversification model whose rates (r = spec. - ext.; a = spec./ext.) depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</li>
<li>BiSSE Speciation/Extinction Likelihood  &#8212; Calculates likelihood with a tree of a species diversification model whose speciation and extinction rates depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</li>
<li>BiSSE Speciation/Extinction Likelihood Calculator  &#8212; Calculates likelihood with a tree of a species diversification model whose speciation and extinction rates depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</li>
<li>BiSSE Trees &amp; Characters  &#8212; Generates tree by a speciation/extinction model in which a character controls rates of speciation/extinction.  Time to next event drawn from negative exponential distribution.</li>
<li>BLAST in Web Browser  &#8212; BLASTs selected data against GenBank.</li>
<li>Boolean for Character (in List of Characters window)  &#8212; Supplies booleans for characters for a character list window.</li>
<li>Boolean for Tree (in List of Trees window)  &#8212; Supplies booleans for trees for a trees list window.</li>
<li>Boolean Info Strip  &#8212; Shows an strip of boxes near the top of each a character column that shows the value of a boolean for that character in gray or white.</li>
<li>Boolean Tree Value  &#8212; Gives a tree a value of 1 if it satisfies the criterion, 0 otherwise.</li>
<li>Bootstrap Resample Matrix  &#8212; Resamples characters from a matrix for use in bootstrap methods.</li>
<li>Branch Information  &#8212; Provides a tool that shows information about branches.</li>
<li>Branch Length from Root  &#8212; For each taxon, calculates the sum of the branch lengths from that taxon to the root of the tree.</li>
<li>Branch Lengths Adjust  &#8212; Provides a tool to adjust branch lengths of trees.</li>
<li>Branch Lengths from current, reinterpreted as Divergence Times &mdash; Reassigns branch lengths under the assumption that currently assigned branch lengths are actually divergence times.</li>
<li>Brownian Model  &#8212; Initializes Brownian motion character model for likelihood and other probability calculations.</li>
<li>Calculated value for taxon  &#8212; Supplies values for taxa reinterpreted as if characters.</li>
<li>Canonical Variates Analysis  &#8212; Performs canonical variates analysis on a continous-valued matrix.</li>
<li>Catalogue of Life  &#8212; Provides a URL to a taxon in the Catalogue of Life</li>
<li>Categorical Data Package Introduction  &#8212; Provides basic facilities for handling categorical characters.</li>
<li>Cell Wand (data)  &#8212; Provides a tool with which to select cells in a matrix automatically.</li>
<li>Character Compositional Bias  &#8212; Calculates the percent of taxa with particular nucleotides (GC bias, or individual frequency of A, C, G or T) for a character.</li>
<li>Character Group Color  &#8212; Colors by Character Group.</li>
<li>Character Included  &#8212; A boolean that is true if a character is currently included, and false if the character is currently excluded.</li>
<li>Character Inclusion  &#8212; Shows current character inclusion in character list window.</li>
<li>Character Likelihood  &#8212; Calculates the negative log likelihood of a tree for a given character.</li>
<li>Character List  &#8212; Makes windows listing characters and information about them.</li>
<li>Character Loadings  &#8212; Returns loadings of a character in an ordination</li>
<li>Character Matrices List  &#8212; Makes windows listing character data matrices and information about them.</li>
<li>Character Matrices Package Introduction  &#8212; Provides basic management and utilities for character data matrices.</li>
<li>Character Matrix Editor  &#8212; Makes editor windows to edit character data.</li>
<li>Character Matrix manager  &#8212; Coordinates the management of character data matrices.</li>
<li>Character matrix selection coordinator  &#8212; Coordinates selecting taxa and characters in data matrix.</li>
<li>Character Models List  &#8212; Makes windows listing character models and information about them.</li>
<li>Character Reference Strip  &#8212; Shows the strip at bottom of matrix with character reference.</li>
<li>Character selection coordinator  &#8212; Coordinates character selection.</li>
<li>Character Sort (data)  &#8212; Provides a tool with which to sort characters automatically.</li>
<li>Character Source   &#8212; Coordinates the supply of characters from various sources of characters.</li>
<li>Character Source  &#8212; Coordinates the supply of characters from various sources of characters.</li>
<li>Character value using other character  &#8212; Coordinates the calculation of a number for a character based on another character (e.g., a correlation between the two characters).</li>
<li>Character value with current tree  &#8212; Coordinates the calculation of a number for a character based on a current tree.</li>
<li>Character Value with Respective Tree  &#8212; Coordinates the calculation of a number for a character based on a respective tree.</li>
<li>Character value with tree  &#8212; Coordinates the calculation of a number for a character based on a tree.</li>
<li>Character Wand (data)  &#8212; Provides a magic wand tool with which to select characters automatically.</li>
<li>Character-Associated Diversification  &#8212; Coordinates analyses for the effect of a character on diversification (speciation/extinction).</li>
<li>Character/State Information  &#8212; Exports character and state information for categorical matrix.</li>
<li>Characters from Matrix Source  &#8212; Supplies characters from source of matrices.</li>
<li>Characters from Ordinations  &#8212; Supplies a character from an ordination of an existing matrix.</li>
<li>Characters Scattergram  &#8212; Makes a scatterplot displaying two values (such as likelihoods, parsimony scores, etc.) for a series of characters.</li>
<li>Chart from Instruction File  &#8212; Displays a chart summarizing data contained in files, as directed by an instruction file.</li>
<li>Charting Package Introduction  &#8212; Provides basic charting functions.</li>
<li>Circular tree  &#8212; Draws trees in circular form.</li>
<li>Clade Match with Another Current Tree  &#8212; Calculates a number relating each clade in the tree to another current tree (e.g., a tree-to-tree distance metric).</li>
<li>Clear All Node Names  &#8212; Clears all node names from the tree.</li>
<li>Clustal (DNA/RNA)  &#8212; Imports Clustal files that consist of DNA or RNA sequence data.</li>
<li>Clustal (protein)  &#8212; Imports Clustal files that consist of amino acid sequence data.</li>
<li>ClustalW Align  &#8212; Sends the selected sequence to ClustalW to align.</li>
<li>Cluster analysis  &#8212; Supplies trees obtained from cluster analysis on distance matrices.</li>
<li>Coalescence Contained within Current Tree  &#8212; Generates tree by a simple coalescence model of a neutral gene with constant population size, within a current species tree from a Tree window or other tree context. Branch lengths are assigned according to generation of coalescence.The species tree used is a current tree found in a Tree Window or other tree context.</li>
<li>Coalescence in Current Tree with Migration  &#8212; Generates tree by a simple coalescence model of a neutral gene with constant population size, within a current species tree from a Tree window or other tree context. Branch lengths are assigned according to generation of coalescence.  The default population size is 10000. The species tree used is a current tree found in a Tree Window or other tree context. Migration is allowed; the default probability is 0.00001 per individual generation. Migration can be isolated to a particular generation, or spread throughout the depth of the species/population history.</li>
<li>Coalescence Package Introduction  &#8212; Provides simulations and calculations for coalescence (genetic drift) within populations.</li>
<li>Coalescence Simulated within Current Tree  &#8212; Generates tree by a simple coalescence model of a neutral gene with constant population size, within species trees. Branch lengths are assigned according to generation of coalescence.The species tree used is a current tree found in a Tree Window or other tree context.</li>
<li>Coalescent Trees  &#8212; Generates tree by coalescence within a single panmictic population.</li>
<li>Codon Position Rates Model  &#8212; Defines and maintains models that specify rates for codon positions.</li>
<li>Collapse Gaps to Right  &#8212; Collapses gaps in the selected block of cells to the right, to yield unaligned sequences.</li>
<li>Collapse Sequences to Left (Remove all Gaps)  &#8212; Collapses sequences in the selected block of cells to yield unaligned sequences, by removing all gaps.</li>
<li>Collapse Zero-length Branches  &#8212; Collapses zero-length branches to yield polytomes.</li>
<li>Colless's Imbalance  &#8212; Calculates Colless's Imbalance statistic for tree asymmetry, normalized by maximum asymmetry.  Polytmous nodes are ignored in the calculation.</li>
<li>Color Branches  &#8212; Provides a tool with which to color branches in a tree window.</li>
<li>Color Branches by Partition  &#8212; Colors the tree's branches by the taxon partition.</li>
<li>Color By Amino Acid  &#8212; Colors the cells of a DNA matrix by the amino acids for which they code.</li>
<li>Color By Cell Value  &#8212; Colors the cells of a character matrix according to a value for that cell or a moving window of cells.</li>
<li>Color By Character Value  &#8212; Colors the cells of a character matrix according to a value for the character or a moving window of characters.</li>
<li>Color By Footnote  &#8212; Colors the cells of a character matrix by whether or not they have a footnote.</li>
<li>Color By State  &#8212; Colors the cells of a character matrix by their contained character states.</li>
<li>Color By Taxon Value  &#8212; Colors the cells of a character matrix according to a value for the Taxon or a moving window of taxa.</li>
<li>Color For Aligning  &#8212; Colors the cells of a character matrix for easy of alignment.</li>
<li>Color If Excluded  &#8212; Colors the excluded characters gray.</li>
<li>Color Taxa  &#8212; Provides paintbrush to color taxon names.</li>
<li>Compare Bases with Taxon  &#8212; Compares nucleotide bases of selected taxa with those of a chosen taxon.</li>
<li>Compare Other Matrix  &#8212; Compares matrix in data editor with another.</li>
<li>Compare with Another Current Tree  &#8212; Calculates a number relating a tree and another current tree (e.g., a tree-to-tree distance metric).</li>
<li>Compare with Other Trees  &#8212; Calculates a number relating one tree to another in a sequence (e.g., a tree-to-tree distance metric), such that the i'th tree is compared to the i'th other tree in the sequence.</li>
<li>Composite DNA Simulation Model  &#8212; Defines and maintains simple stochastic models of DNA evolution for simulations.</li>
<li>Concatenate Multiple Tree Sources  &#8212; Concatenates multiple tree sources to yield a tree source supplying trees from each source.  The trees are marked with values to indicate the source.</li>
<li>Concatenate Other Matrix  &#8212; Concatenates matrix onto one in data editor.  Assumptions like weights and character models are NOT transferred.  For categorical data, state names are not included.  For continuous data, new items may need to be created to accommodate differences in items between the matrices.</li>
<li>Concatenate Respective Matrices from Two Sources  &#8212; Supplies matrices by concatenating respective matrices from two sources.  The i'th matrix supplied is the concatenation of the i'th matrix from each of the two sources.  Assumptions like weights and character models are NOT transferred.  For categorical data, state names are not included.  For continuous data, new items may need to be created to accommodate differences in items between the matrices.</li>
<li>Concatenate tree blocks  &#8212; Concatenates selected tree blocks into a new tree block.</li>
<li>Concatenate Two Tree Sources  &#8212; Concatenates two tree sources to yield a tree source first supplying trees from one source, then supplying trees from another.  The trees are marked with values 1 and 2 to indicate the source.</li>
<li>Condense taxon names  &#8212; Condenses taxon names, for example for use by programs that cannot handle long names or names with punctuation.</li>
<li>Condensed Tree Source  &#8212; Produces a block of trees that are a condensed version of another block of trees, with duplicate trees removed.</li>
<li>Consensus Sequence Strip  &#8212; Displays a consensus sequence as in info strip in a character matrix editor.</li>
<li>Consensus State for Character  &#8212; </li>
<li>Consensus Tree  &#8212; Supplies consensus tree from a block of trees.</li>
<li>Consensus Trees from Tree Blocks  &#8212; Supplies consensus trees from Tree blocks.</li>
<li>Consensus Trees Package Introduction  &#8212; Calculations for consensus trees.</li>
<li>Consistency Index for Character  &#8212; Calculates the Consistency Index (CI) for a tree and character.</li>
<li>Consistency Index for Matrix   &#8212; Calculates the Consistency Index (CI) for a tree and matrix.</li>
<li>Constellation  &#8212; Draws trees with spots at the nodes in a more or less unrooted way.</li>
<li>Contained Gene (or Other) Trees  &#8212; Draws wide trees containing trees of associates (e.g., species tree containing gene tree)</li>
<li>Continuous Data Package Introduction  &#8212; Provides basic facilities for handling continuous-valued characters.</li>
<li>Continuous Items Editor  &#8212; Edits items for continuous data matrix within the editor info panel.</li>
<li>Continuous state of taxon  &#8212; State of a continuous character.</li>
<li>Convert Negative Branch Lengths to Zero  &#8212; Adjusts a tree's branch lengths so that negative branch lengths are zero</li>
<li>Convert Polymorphisms to Uncertainties  &#8212; Alters categorical data by converting polymorphic entries to uncertainties.</li>
<li>Convert to RY  &#8212; Converts nucleotide data to RY data in selected region.</li>
<li>Convert Uncert./Polymorph. to Missing  &#8212; Alters categorical data by converting uncertain or polymorphic entries to missing data.</li>
<li>Convert Uncertainties to Polymorphisms  &#8212; Alters categorical data by converting uncertain entries to polymorphisms.</li>
<li>Correl Package  &#8212; Correl is a Mesquite package for examining character correlation.</li>
<li>Correlation Analysis  &#8212; Performs a Correlation analysis for categorical characters.</li>
<li>Current Codon Positions  &#8212; Supplies current codon positions applied to characters for character list window.</li>
<li>Current Genetic Codes   &#8212; Shows current genetic codes applied to characters in character list window.</li>
<li>Current Genetic Codes  &#8212; Supplies the currently assigned genetic code (e.g., "standard") for a character.</li>
<li>Current Parsimony Models   &#8212; Shows current parsimony models applied to characters in character list window.</li>
<li>Current Parsimony Models  &#8212; Supplies the currently assigned parsimony model (e.g., "unordered") for a character.</li>
<li>Current Probability Models  &#8212; Supplies current stochastic models applied to characters for character list window.</li>
<li>Current probability models  &#8212; Supplies the currently assigned stochastic (likelihood, probability) model for a character.</li>
<li>Current taxa partition  &#8212; Supplies the current taxa partition.</li>
<li>Current Tree  &#8212; Supplies a single tree currently shown in a tree window.</li>
<li>Current Weights  &#8212; Supplies current weights applied to characters for character list window.</li>
<li>Curvogram  &#8212; Draws trees with curved branches (as PHYLIP's 'Curvogram').</li>
<li>Data Painter  &#8212; Supplies the paint can tool and dropper tool for filling cells in a character data editor.</li>
<li>Data type for model  &#8212; Indicates data type of model in list window.</li>
<li>Data Window Coordinator  &#8212; Coordinates the creation of basic data windows.</li>
<li>Deassign Branch Lengths  &#8212; Sets lengths of a tree's branches to unassigned.</li>
<li>Deep Coalescences (gene tree)  &#8212; Counts W. Maddison's (1997) number of extra gene lineages ("deep coalescences") for gene tree within species tree</li>
<li>Deep Coalescences (species tree)  &#8212; Counts W. Maddison's (1997) number of extra gene lineages ("deep coalescences") for species tree implied by a contained gene tree</li>
<li>Deep Coalescences Multiple Loci  &#8212; Counts W. Maddison's (1997) number of extra gene lineages ("deep coalescences") for species tree containing a series of gene trees.  The gene trees must be represented by a single taxa block.</li>
<li>Default Order (characters)  &#8212; Shows default order of character.</li>
<li>Default Order (taxa)  &#8212; Shows default order of taxa.</li>
<li>Default Trees  &#8212; Supplies simple default trees (bush, ladder).</li>
<li>Defaults  &#8212; Supervises some Mesquite-wide defaults</li>
<li>Definite Tree Source  &#8212; Supplies trees from from a tree source, arranged to be a definite number</li>
<li>Delete Reticulations  &#8212; Deletes secondary parents of nodes with more than one immediate ancestor.  There is no control over which parents get deleted.</li>
<li>Dependent Contained Tree Window  &#8212; Displays a single tree (the same as contained within a tree in a tree window).</li>
<li>Dependent Tree Window  &#8212; Displays a single tree (the same as in a tree window).</li>
<li>Diagonal tree  &#8212; Draws trees with standard diagonal branches ("cladogram")</li>
<li>Difference in steps in two characters  &#8212; Calculates the difference between two characters in parsimony steps for a given tree.</li>
<li>Difference in two values for character  &#8212; Coordinates the calculation of the difference in two numbers for a character.</li>
<li>Difference in two values for tree  &#8212; Coordinates the calculation of the difference in two numbers for a tree.</li>
<li>Display Character Loadings  &#8212; Shows character loadings in ordination.</li>
<li>Display Mean  &#8212; Calculates and shows the mean value.</li>
<li>Distance between taxa  &#8212; Counts the difference between two taxa.</li>
<li>Distance Matrix Package Introduction  &#8212; Calculates and provides utilities for distances matrices.</li>
<li>Distance of Contained Taxa  &#8212; Distances among taxa according to distances among contained taxa (e.g., genes).</li>
<li>Distances from Character Matrix  &#8212; Distances calculated from a character matrix.</li>
<li>Diverse Package  &#8212; Diverse is a Mesquite package for examining speciation and extinction.</li>
<li>Diversification (Char. Indep.)  &#8212; Coordinates analyses of diversification (speciation/extinction).</li>
<li>DManager Package Introduction  &#8212; Provides special tools for managing data and taxa.</li>
<li>DNA Empirical State Frequencies Model  &#8212; Defines and maintains model of state frequences that match those found in an empirical matrix.</li>
<li>DrawHierarchy  &#8212; Draws project and module trees</li>
<li>E-mail Tree  &#8212; Calls mailto: to email a text image of the tree.</li>
<li>Edit State Names  &#8212; Edits state names of a categorical data matrix.</li>
<li>Enforce Minimum Node Age Constraints &#8212; Uses minimum node age constraints to set the branch lengths on the tree.</li>
<li>Equiprobable Trees  &#8212; Generates trees randomly so that each possible labelled tree topology is equally likely.</li>
<li>Evolutionary Principal Components  &#8212; Performs Maddison and Dyreson's evolutionary principal components analysis on a continous-valued matrix.</li>
<li>Evolve Categorical Characters  &#8212; Simulates character evolution for categorical characters on a tree.</li>
<li>Evolve Continuous Characters  &#8212; Simulates character evolution for continuous characters on a tree.</li>
<li>Evolve DNA Characters  &#8212; Simulates evolution of DNA sequences on a tree.</li>
<li>Evolving Speciation Rate (Continuous Character)  &#8212; Generates tree by a speciation model in which the speciation rate evolves by a Brownian motion model.</li>
<li>Examples Navigator  &#8212; Provides a Examples Navigator window with explanation and buttons to link to other files</li>
<li>Export Ancestral States Trace &mdash; Exports the ancestral state trace from Trace Character History.</li>
<li>Export Matrices &amp; Batch Files  &#8212; Saves copies of matrices to separate files for subsequent batch analysis (e.g., parametric bootstrapping).</li>
<li>Export NEXUS for MrBayes  &#8212; Exports NEXUS files for use by MrBayes.</li>
<li>Export NEXUS Tree File  &#8212; Exports NEXUS file with a tree block, and optionally a taxa block.</li>
<li>Export NEXUS Tree File from Tree Source  &#8212; Exports NEXUS file with a tree block, and optionally a taxa block, based upon a source of trees.  One major advantage of this is that it allows a collection of trees to a file without having them all in memory at once. For example, if the Tree Source is "Transform Trees From Other Source", and the other source is "Use Trees from Separate NEXUS file", then Mesquite will read in a tree from the other NEXUS file, transform the  tree, and wrte it to the file, one at a tme.</li>
<li>Export Partitions as Separate NEXUS files &#8212; Exports character partitions in a matrix as Separate NEXUS files.</li>
<li>Export Taxa Distance Matrix  &#8212; Exports a tab-delimited text file of distances among the taxa.</li>
<li>Extract tree block  &#8212; Extracts selected trees and places them in a new tree block.</li>
<li>F81 distance  &#8212; F81 (Felsenstein, 1981) distance from a character matrix.</li>
<li>F84 distance  &#8212; F84 (Felsenstein, 1984) distance from a DNA matrix.</li>
<li>FASTA (DNA/RNA)  &#8212; Imports and exports FASTA files that consist of DNA/RNA sequence data.</li>
<li>FASTA (protein)  &#8212; Imports and exports FASTA files that consist of amino acid sequence data.</li>
<li>Fetch &amp; Add GenBank Sequences  &#8212; Fetches GenBank nucleotide sequences given their GenBank accession numbers and adds them to the matrix.</li>
<li>File Comment  &#8212; Manages the main file comment in a NEXUS file.</li>
<li>Fill  &#8212; Fills cells with a uniform state in a character data editor.</li>
<li>Filter Trees from Other Source  &#8212; Filters trees from another source.</li>
<li>Find Sequence  &#8212; Finds the next occurrence of a sequence in a matrix of molecular data.</li>
<li>Forward/Backward Rates  &#8212; Uses maximum likelihood to estmate the rates of forward and backward changes (0 to 1 and 1 to 0 changes respectively), or alternatively the overall rate and the bias in gains versus losses, using the AsymmMk model on a tree for a given character.</li>
<li>Fraction Applicable and Not Missing  &#8212; Calculates the fraction of the matrix that is applicable and not missing data.</li>
<li>Frequency of State  &#8212; Calculates the frequency of a state for a categorical character</li>
<li>Fused Matrix Export (NEXUS)  &#8212; Exports NEXUS files with matrices fused.</li>
<li>Gamma Invar Rates Model  &#8212; Defines and maintains Gamma rate variation models with a proportion of invariant characters.</li>
<li>Gamma Rates Model  &#8212; Defines and maintains Gamma rate variation models.</li>
<li>Gaps to Missing  &#8212; Converts gaps to missing data in selected region.</li>
<li>GC  &#8212; Returns 1 for G or C, 0 for A or T.</li>
<li>GenBank (DNA/RNA)  &#8212; Imports GenBank files that consist of DNA/RNA sequence data.</li>
<li>GenBank Number &#8212; Lists the primers used in the reads composing a sequence.</li>
<li>GenBank Taxon  &#8212; Provides a URL to a taxon in NCBI GenBank</li>
<li>Gene Duplication-Extinction (gene tree)  &#8212; Counts the number of duplication and extinction events for gene tree implied by a containing species tree</li>
<li>Gene Duplication-Extinction (species tree)  &#8212; Counts the number of duplication and extinction events for species tree implied by a contained gene tree</li>
<li>Genesis Introduction  &#8212; Implements stochastic models of molecular evolution.</li>
<li>GenPept (Protein)  &#8212; Imports GenPept files that consist of amino acid sequence data.</li>
<li>Get Taxon's tree from a web site  &#8212; Supplies a tool for tree windows that gets tree for taxon touched from a web site.</li>
<li>Get Taxon's tree from tolweb.org  &#8212; Supplies a tool for tree windows that gets tree for taxon touched from tolweb.org</li>
<li>Go-to Window  &#8212; Provides a window to jump to another file</li>
<li>Graft Other Tree &#8212; Grafts a tree in a tree window onto given tree; requires that the other tree includes none of the same terminal taxa.  If a taxon in receiving tree is selected, graft occurs there; otherwise, graft occurs at root.</li>
<li>Group Membership (characters)  &#8212; Lists and allows changes to group membership in the current partition of characters, for List of Characters window.</li>
<li>Group Membership (taxa)  &#8212; Lists and allows changes to group membership in the current partition of taxa, for List of Taxa window.</li>
<li>GTR Rate Matrix Model  &#8212; Defines and maintains six-parameter general time-reversible (GTR) rate matrix models.</li>
<li>Highlight Apparently Slightly Misaligned  &#8212; Colors aligned sequences to emphasize sections that seem slightly misaligned.</li>
<li>Import and Export Package Introduction  &#8212; Provides file import/export utilities.</li>
<li>Initialize DNA Submodels  &#8212; Initializes default DNA submodels.</li>
<li>Initialize Genetic Code  &#8212; Initializes default genetic codes.</li>
<li>Initialize Parsimony  &#8212; Initializes default character models for parsimony.</li>
<li>Initialize predefined categorical parsimony models  &#8212; Initializes the predefined categorical parsimony models.</li>
<li>Initialize predefined continuous parsimony models  &#8212; Initializes the predefined continuous parsimony models.</li>
<li>Initialize predefined meristic parsimony models  &#8212; Initializes the predefined meristic parsimony models.</li>
<li>Insert Node  &#8212; Provides a tool with which to insert nodes along a branch.</li>
<li>Installer  &#8212; Installs packages.</li>
<li>Interpolate Character Selection  &#8212; Selects characters that are surrounded by characters selected.</li>
<li>iSpecies  &#8212; Provides a URL to a taxon in iSpecies</li>
<li>Item values chart  &#8212; Manages chart of values for items.</li>
<li>Items bi-plot  &#8212; Makes a chart comparing two values for each of many items.</li>
<li>iText  &#8212; iText is a package for saving PDF files within Mesquite.</li>
<li>JAMA Introduction  &#8212; Introduces the JAMA library.</li>
<li>Jukes Cantor distance  &#8212; Jukes Cantor distance from a DNA matrix.</li>
<li>K2P distance  &#8212; K2P (Kimura 2-parameter) distance from a character matrix.</li>
<li>Keep Prefix/Suffix of Taxon Name  &#8212; Keeps only Prefix or Suffix of Taxon Name.</li>
<li>Kyte &amp; Doolittle Hydrophobicity  &#8212; Returns hydrophobicity of amino acid using the scale of Kyte, J &amp; R.F. Doolittle (1982) J. Mol. Biol. 157:105-142.  Numbers from http://www.whatislife.com/reader/protein/aa.html</li>
<li>Label Branch Lengths  &#8212; An assistant to a tree display that labels branches to show their lengths.</li>
<li>Label states  &#8212; A module that displays character states on tree using labels.  This is a display-only module, and would be hired by another module that organizes assigning states to the nodes.</li>
<li>Landmark Drawings  &#8212; Places drawing of landmarks at each node in tree.</li>
<li>Likelihood Ancestral States  &#8212; Coordinates reconstruction of ancestral states by maximum likelihood.  Currently each node is estimated independently (i.e., corresponding to PAUP's marginal reconstruction).</li>
<li>Likelihood in Character  &#8212; Calculates the negative log likelihood of a tree with respect to a single character.</li>
<li>Likelihood Surface AsymmMk Model  &#8212; Shows the likelihood surface for the AsymmMk Model.</li>
<li>Lineages Through Time  &#8212; Displays a window showing plot of lineages through time averaged over many trees.</li>
<li>List of Character Inclusion Sets  &#8212; Makes windows listing character sets.</li>
<li>List of Character Partitions  &#8212; Makes windows listing character partitions.</li>
<li>List of Character Sets  &#8212; Makes windows listing character sets.</li>
<li>List of Character Weight Sets  &#8212; Makes windows listing character weight sets.</li>
<li>List of Codon Positions Sets  &#8212; Makes windows listing codon positions sets.</li>
<li>List of Genetic Code Sets  &#8212; Makes windows listing genetic code sets.</li>
<li>List of Parsimony Model Sets  &#8212; Makes a window listing parsimony model sets.</li>
<li>List of Probability Model Sets  &#8212; Makes windows listing probability model sets.</li>
<li>List of Taxa Partitions  &#8212; Makes windows listing taxa partitions.</li>
<li>List of Taxon Sets  &#8212; Makes windows listing taxon sets.</li>
<li>List Windows Package Introduction  &#8212; Provides List windows to manage taxa, characters and other information.</li>
<li>Log transform  &#8212; Alters continuous data by log transforming values.</li>
<li>Lower Case  &#8212; Returns 1 if state symbol is a lower case letter.</li>
<li>Maintain Clipboard Match  &#8212; Finds the first occurrence of the sequence in  the clipboard, within a designated taxon, and maintains that match as the clipboard changes.</li>
<li>Maintain Target Match  &#8212; Finds the first occurrence of the sequence in  the text area below the matrix, within a designated taxon, and maintains that match as the text area changes.</li>
<li>Majority Rules Consensus  &#8212; Calculates the majority rules consensus tree.</li>
<li>Make Rerootings of Clade  &#8212; Presents a tool by which you can touch on a tree;  a tree block is made consisting of trees representing all rerootings of the clade of the node touched.</li>
<li>Manage Alternative Taxon Names  &#8212; Manages (including NEXUS read/write) alternative taxon names.</li>
<li>Manage ASSUMPTIONS blocks  &#8212; Manages ASSUMPTIONS block in NEXUS file.</li>
<li>Manage Attached Notes  &#8212; Manages (including NEXUS read/write) notes attached to taxa, characters and cells of matrices.</li>
<li>Manage AUTHORS blocks  &#8212; Manages AUTHORS block in NEXUS file.</li>
<li>Manage categorical character matrices  &#8212; Manages categorical data matrices (including read/write in NEXUS file).</li>
<li>Manage character inclusion sets  &#8212; Manages (including NEXUS read/write) character inclusion sets (EXSETS).</li>
<li>Manage Character Models  &#8212; Manages character models.</li>
<li>Manage character partititions  &#8212; Manages (including NEXUS read/write) character partitions.</li>
<li>Manage character sets  &#8212; Manages (including NEXUS read/write) character sets (CHARSETS).</li>
<li>Manage character weight sets  &#8212; Manages (including NEXUS read/write) character weight sets.</li>
<li>Manage codon positions  &#8212; Manages (including NEXUS read/write) codon position sets.</li>
<li>Manage CODONS blocks  &#8212; Manages CODONS block in NEXUS file.</li>
<li>Manage Continuous char. matrices  &#8212; Manages continuous data matrices (including read/write in NEXUS file).</li>
<li>Manage DNA/RNA matrices  &#8212; Manages DNA/RNA data matrices (including read/write in NEXUS file).</li>
<li>Manage Foreign Blocks  &#8212; Manages unrecognized blocks in a NEXUS file.</li>
<li>Manage genetic code sets  &#8212; Manages (including NEXUS read/write) genetic code sets (CODESETs).</li>
<li>Manage Geographic character matrices  &#8212; Manages data matrices of geographic data (including read/write in NEXUS file).</li>
<li>Manage hyperlinks  &#8212; Manages (including NEXUS read/write) hyperlinks.</li>
<li>Manage LABELS blocks  &#8212; Manages LABELS block in NEXUS file.</li>
<li>Manage Meristic char. matrices  &#8212; Manages meristic data matrices (including read/write in NEXUS file).</li>
<li>Manage MESQUITE block  &#8212; Manages MESQUITE block (including compose snapshot of current state of file in MESQUITE block "Auto" in NEXUS file).</li>
<li>Manage MESQUITECHARMODELS blocks  &#8212; Manages MESQUITECHARMODELS block in NEXUS file.</li>
<li>Manage NOTES blocks  &#8212; Manages footnotes and pictures attached to taxa, characters and data points (including read/write NOTES block in NEXUS file).</li>
<li>Manage parsimony model sets  &#8212; Manages (including NEXUS read/write) parsimony model sets (TYPESETs).</li>
<li>Manage pictures  &#8212; Manages (including NEXUS read/write) pictures.</li>
<li>Manage probability model sets  &#8212; Manages (including NEXUS read/write) probability model sets (PROBMODELSETs).</li>
<li>Manage Protein matrices  &#8212; Manages Protein data matrices (including read/write in NEXUS file).</li>
<li>Manage SETS blocks  &#8212; Manages character sets and other sets of various kinds (including read/write SETS block in NEXUS file).</li>
<li>Manage stored character orders  &#8212; Manages (including NEXUS read/write) character orders.</li>
<li>Manage TAXA blocks  &#8212; Manages sets of taxa (including read/write TAXA block in NEXUS file).</li>
<li>Manage taxa partititions  &#8212; Manages (including NEXUS read/write) taxa partitions.</li>
<li>Manage TaxaAssociation blocks  &#8212; Manages TaxaAssociation blocks in NEXUS file.</li>
<li>Manage taxon sets  &#8212; Manages (including NEXUS read/write) taxon sets.</li>
<li>Manage TREES blocks  &#8212; Manages tree blocks (including read/write TREES block in NEXUS file).</li>
<li>Map Continuous  &#8212; Supplies a reconstruction of continuous-valued ancestral states on a tree. Differs from the basic reconstruction methods in that it supplies simply numbers for nodes, instead of a CharacterHistory</li>
<li>Matching Sequence  &#8212; Finds the next occurrence of a sequence in a matrix of molecular data.  Allows a certain number of mismatches.</li>
<li>Matrices from Ordinations  &#8212; Supplies character matrices from ordinations of existing matrices.</li>
<li>Matrix from Clipboard  &#8212; Makes a character matrix from text in the clipboard.</li>
<li>Matrix Info  &#8212; Shows Information about the data in each taxon.</li>
<li>Matrix Source (indep.)  &#8212; Coordinates the supply of matrices to other modules.  It acts independently, in that it decides what matrix to supply (e.g., by presenting an interface to choose).</li>
<li>Matrix Source (obed.)  &#8212; Coordinates the supply of matrices to other modules.  It acts obediently, in that it does not present an interface to choose which matrix, but relies on employer to decide.</li>
<li>Matrix value with current tree  &#8212; Coordinates the calculation of a number for a character matrix based on a current tree.</li>
<li>Maximum likelihood reconstruct (Generic categorical)  &#8212; Assesses likelihood for categorical characters, and r econstructs ancestral states by the maximum marginal probability ("MLE") criterion</li>
<li>Mean value of character  &#8212; Calculates the mean observed trait for continuous character.</li>
<li>Mean value of character (linked matrix)  &#8212; Calculates the mean observed trait for continuous character in a linked matrix.</li>
<li>Merge Matrices from File &#8212; Includes a file and concatenates its taxa to an existing taxa block, for instance to add seqences.</li>
<li>Merge Taxa  &#8212; Merges selected taxa and their character states.</li>
<li>Merge Taxa &amp; Matrices from File  &#8212; Includes a file and concatenates its taxa to an existing taxa block, for instance to add seqences.</li>
<li>Meristic Data Package Introduction  &#8212; Provides basic facilities for handling meristic characters.</li>
<li>Meristic Items Editor  &#8212; Edits items for meristic data matrix within the editor info panel.</li>
<li>Mesquite  &#8212; This is the central module for the Mesquite system. In it is the main method that starts up the Mesquite application.  This moduleloads information for all of the other modules, and hires FileCoordinator modules as needed to deal with open files.  Thus, in the tree of employeesthat active modules make, this module is at its root. (null, version 2.72)</li>
<li>Mesquite Heuristic (Add &amp; rearrange)  &#8212; Searches for optimal trees by adding taxa then rearranging the tree.</li>
<li>mesquite.jsci Package Introduction  &#8212; Introduces the JSci library.</li>
<li>Mirror Tree Window  &#8212; Displays a single tree (the same as in a tree window) twice, in mirror image.</li>
<li>Missing to Gaps  &#8212; Converts missing data to gaps in selected region.</li>
<li>Mk1 Estimated Rate  &#8212; Estimates the rate of a character's evolution under the simple Mk1 model.</li>
<li>Mk1 Model (Markov 1 parameter)  &#8212; Defines and maintains simple Markov k-state 1-parameter stochastic models (Lewis, 2001) of character evolution.</li>
<li>Model Paradigm  &#8212; Indicates paradigm of model in list window.</li>
<li>Modify Current Tree  &#8212; Modifies current tree.</li>
<li>Molecular Utilities Package Introduction  &#8212; Provides utilities for molecular data.</li>
<li>Most Pairs  &#8212; Chooses taxon pairings regardless of contrast in a character, so as to maximize the number of pairs that are phylogenetically independent.</li>
<li>Move block  &#8212; Allows one to manually blocks in a sequence, and split the blocks.</li>
<li>Move Prefix/Suffix of Taxon Name  &#8212; Moves the suffix of the taxon name to be to prefix or the prefix to be the suffix</li>
<li>MrBayes Max. A-Post. Tree  &#8212; Supplies tree from MrBayes with highest posterior probability (MAP tree).</li>
<li>MrBayes Package Introduction  &#8212; Provides links to MrBayes analyses.</li>
<li>MrBayes Score  &#8212; Supplies posterior probability score from MrBayes</li>
<li>MrBayes Trees from .t File  &#8212; Supplies trees from a ".t" file produced by MrBayes</li>
<li>MRP Matrices from Trees  &#8212; Supplies matrices which represent trees for MRP (Matrix Representation with Parsimony) supertree analyses.</li>
<li>Multi Tree Window  &#8212; Displays a special tree window with many trees simultaneously.</li>
<li>Multiply by value  &#8212; Alters continuous data by multiplying by a value.  All items of the matrix are similarly modified.</li>
<li>Multistate sites in taxon  &#8212; Reports the number of polymorphic/partially uncertain sites in a taxon for a data matrix.</li>
<li>Muscle Align  &#8212; Sends the selected sequence to Muscle to align.</li>
<li>NBRF/PIR (DNA/RNA)  &#8212; Imports and exports NBRF files that consist of DNA/RNA sequence data.</li>
<li>NBRF/PIR (protein)  &#8212; Imports and exports NBRF files that consist of amino acid sequence data.</li>
<li>Neutral Coalescence  &#8212; Performs neutral coalescence within a population by a simple model of a neutral gene with constant population size.Remembers a tree of nodes, with branch lengths assigned according to generation of coalescence.  The default population size is 10000.</li>
<li>NEXUS Blocks List  &#8212; Makes windows listing NEXUS blocks.</li>
<li>NEXUS Defaults  &#8212; Sets whether TITLE and LINK commands are to be suppressed where possible in saving NEXUS files.</li>
<li>NEXUS file interpreter  &#8212; Coordinates the reading and writing of NEXUS files.</li>
<li>NNI Rearranger  &#8212; Rearranges a tree by nearest neighbor interchange (NNI).  Does not handle trees with polytomies.</li>
<li>No Color  &#8212; Turns off cell coloring.</li>
<li>Node Age Constraints  &#8212; Supplies a tool for tree windows to set node age constraints.  A short text string summarizes constraints.  E.g. to say minimum age is 2.0, enter "2.0+".  To say range is 2.0 to 4.0, enter "2.0-4.0".  To say maximum is 4.0, enter "0.0-4.0".  To say age is exactly 3.0, enter "3.0"</li>
<li>Node Count (Speciation Patristic) Distances implied by Tree  &#8212; Distances among taxa implied by counting nodes along branches of tree from one taxon to another (Speciation Patristic distances).</li>
<li>Node Depth  &#8212; Calculates the depth of each node from the highest terminal in its clade.</li>
<li>Node Locations (2D plot)  &#8212; Calculates the node locations for a tree plotted in a two dimensional space.</li>
<li>Node Locations (3D plot)  &#8212; Calculates the node locations for a tree plotted in a three dimensional space.</li>
<li>Node Locations (circle)  &#8212; Calculates the node locations for a tree drawn in circular fashion, with root at center.</li>
<li>Node Locations (oval)  &#8212; Calculates the node locations for a tree drawn in oval fashion, with root at center.</li>
<li>Node Locations (standard)  &#8212; Calculates the node locations in a tree drawing, for use with vertical or horizontal tree drawers (e.g., the standard diagnonal or square trees).</li>
<li>Node Namer  &#8212; Provides a tool to name the nodes of a tree</li>
<li>Node Position Adjust  &#8212; Provides a tool to adjust the depth of a node in the tree</li>
<li>Node-Associated Values  &#8212; Supplies a tool for tree windows to attach and view footnotes for branches.</li>
<li>Nodes Scattergram  &#8212; Shows values for nodes of tree via a scattergram.</li>
<li>NONA, Hennig86, PiWe, WinClada  &#8212; Imports and exports NONA/Hennig86/PiWe/WinClada files.</li>
<li>Nucleotide Complement  &#8212; Alters nucleotide data to its complement.</li>
<li>Number for Character (in List of Characters window)  &#8212; Supplies numbers for characters for a character list window.</li>
<li>Number for Matrix (in List of Character Matrices window)  &#8212; Supplies numbers for character matrcies for a character matrices list window.</li>
<li>Number for Nodes using Character  &#8212; Supplies numbers, based on a character, for each node of a tree.</li>
<li>Number for Nodes using Matrix  &#8212; Supplies numbers, based on a matrix, for each node of a tree.</li>
<li>Number for Taxon (in List of Taxa window)  &#8212; Supplies numbers for taxa for a taxa list window.</li>
<li>Number for Tree (in List of Trees window)  &#8212; Supplies numbers for trees for a trees list window.</li>
<li>Number for Tree Block (in List of Tree Blocks window)  &#8212; Supplies numbers for tree blocks for a tree blocks list window.</li>
<li>Number of Trees Matching Criterion &#8212; Counts the number of trees in a tree block that match a user-specified criterion.  For example, it could count the number of trees in a tree block which are congruent with a particular tree topology (by selecting the "Tree Congruent with Constraint Tree Topology" module in the Criterion for Trees dialog). </li>
<li>Number of characters in character set  &#8212; Indicates number of characters in character set in list window.</li>
<li>Number of characters in inclusion set  &#8212; Indicates number of characters in inclusion set in list window.</li>
<li>Number of characters of data matrix  &#8212; Indicates taxa of data matrix.</li>
<li>Number of Congruent Trees  &#8212; For the given tree, returns the number of trees in the tree source that are congruent with it.</li>
<li>Number of groups in taxa partition  &#8212; Indicates number of groups in taxa partition in list window.</li>
<li>Number of Stops  &#8212; Reports the number of stops coded by nucleotides in a taxon for a data matrix.</li>
<li>Number of Taxa  &#8212; Counts the number of taxa in a tree.</li>
<li>Number of taxa in set  &#8212; Indicates number of taxa in list window.</li>
<li>Number of taxa in taxa set  &#8212; Indicates number of taxa in taxa set in list window.</li>
<li>Number of Taxa with Non-Gap, Non-Missing  &#8212; Calculates the number of taxa in the character that contain applicable, non-missing data.</li>
<li>Number of trees in tree block  &#8212; Indicates number of trees.</li>
<li>Number of Unambiguous Sites in Sequence  &#8212; Reports the number of unambiguous sites.</li>
<li>Numbers for Nodes from Character Reconstructions  &#8212; Supplies numbers, based on a character reconstruction, for each node of a tree.</li>
<li>Obtain Branch Lengths from Numbers For Nodes  &#8212; Assigns a value of branch length for all of a tree's branches based upon the numbers calculated by a Numbers for Nodes.</li>
<li>Obtain  Branch Lengths from Values Attached to Nodes &mdash; Assigns branch lengths to the tree taken from values already attached to nodes, for instance support values or divergence times from other programs.</li>
<li>Occasionally Randomly Modify  &#8212; With specified probabililty, will ask random tree modifier to modify current tree.</li>
<li>Open URL  &#8212; Opens a file on the web as if it were a local data file</li>
<li>Original File (import source)  &#8212; Shows the orignal file from which a fuse/imported taxon came.</li>
<li>Ornamental Trees Package Introduction  &#8212; Has extra methods for drawing and decorating trees.</li>
<li>Other Tree Blocks  &#8212; Supplies blocks of trees from various sources</li>
<li>Pagel format (ppy) file for Discrete  &#8212; Imports and exports files formatted for Pagel's Discrete program.</li>
<li>Pagel format (ppy) file for Multistate  &#8212; Imports and exports files formatted for Pagel's Multistate program.</li>
<li>Pagel's 1994 test of correlated (discrete) character evolution  &#8212; A statistical test, described in Pagel(1994), for nonindependent evolution of two discrete, binary characters</li>
<li>Pairs Contrasting in State of One Character  &#8212; Chooses taxon pairings so as to maximize the number of pairs that are phylogenetically independent, subject to the constraint that each pair shows a contrast the states of a binary character.</li>
<li>Pairs Contrasting in State of Two Characters  &#8212; Chooses taxon pairings so as to maximize the number of pairs that are phylogenetically independent, subject to the constraint that each pair shows a contrast the states of both of two binary characters.</li>
<li>Pairwise Comparison  &#8212; Performs pairwise comparison character correlation tests. Phylogenetically independent pairs are chosen, and the states of two binary characters are compared to see if they are correlated among these pairs.</li>
<li>Pairwise Comparisons Package Introduction  &#8212; Performs pairwise comparisons character correlation analyses.</li>
<li>Pairwise Taxon Difference  &#8212; Calculates distance between a specific pair of taxa based on a character matrix.  Differs from Distance Between Taxa in thatit performs calculation for only a single pair of taxa.</li>
<li>PAL Introduction  &#8212; Introduces the PAL library.</li>
<li>Parameters Explorer  &#8212; Provides a window to show values returned when parameter settings are varied</li>
<li>Parsimony Ancestral States  &#8212; Coordinates the parsimony reconstruction of ancestral states.</li>
<li>Parsimony Character Steps  &#8212; Calculates the number of parsimony steps in a character.</li>
<li>Parsimony Linear  &#8212; Reconstructs the ancestral states of continuous characters so as to minimize the sum of absolute values of changes (linear, Wagner, Farris or Manhattan parsimony).  If the continuous character has multiple items, then length is reported on only the first item.</li>
<li>Parsimony Meristic Linear  &#8212; Reconstructs the ancestral states of mersitic characters so as to minimize the sum of absolute values of changes (linear, Wagner, Farris or Manhattan parsimony).  If the meristic character has multiple items, then length is reported on only the first item.</li>
<li>Parsimony Ordered  &#8212; Reconstructs the ancestral states of categorical characters using parsimony, under the assumption that states are ordered (ordered, Farris or Wagner parsimony; additive).  Also counts parsimony steps.</li>
<li>Parsimony Package Introduction  &#8212; Preforms parsimony calculations to count steps and treelength of characters on trees.</li>
<li>Parsimony Squared-change  &#8212; Reconstructs the ancestral states of continuous characters so as to minimize the sum of squared changes (squared-change or least squares parsimony; Brownian motion likelihood).</li>
<li>Parsimony Stepmatrix  &#8212; Reconstructs ancestral states of categorical characters using a stepmatrix (cost matrix).</li>
<li>Parsimony Unordered  &#8212; Reconstructs the ancestral states of categorical characters using parsimony, under the assumption that states are unordered (unordered or Fitch parsimony; nonadditive).  Also counts parsimony steps.</li>
<li>Patristic distance correlation  &#8212; Calculates the product moment correlation coefficient among the off-diagonal elements of the patristic distances matrices of two trees.  Ignores unassigned and negative values in the matrices.  If employed as a distance, converts scores from 1 to -1 to scores from 0 to 2</li>
<li>Patristic Distances implied by Tree  &#8212; Distances among taxa implied by path-length distance along branches of tree from one taxon to another (Patristic distances).  Unassigned branches are treated as of length 1.0.</li>
<li>Phylip (categorical data)  &#8212; Imports and exports Phylip matrices that consist of basic categorical data with just two states. Exported data will consist of default symbols (0 and 1).</li>
<li>Phylip (DNA/RNA)  &#8212; Imports and exports Phylip matrices that consist of DNA/RNA sequence data.</li>
<li>Phylip (protein)  &#8212; Imports and exports Phylip matrices that consist of amino acid sequence data.</li>
<li>Phylip (trees)  &#8212; Imports and exports Phylip trees.</li>
<li>Picture Window  &#8212; Displays a picture in a window.</li>
<li>Plot Tree  &#8212; Draws trees plotted in a two dimensional space.</li>
<li>Plot Tree 3D  &#8212; Draws trees plotted in a three dimensional space.</li>
<li>Polytomies Present  &#8212; Indicates whether polytomies are present.</li>
<li>Polytomy Assumption  &#8212; Indicates whether the tree assumes that any polytomies, if present, are hard or soft.  If not, the default assumption is used.</li>
<li>POY  (DNA/RNA)  &#8212; Exports POY files of DNA/RNA sequence data.</li>
<li>Principal Components Analysis  &#8212; Performs principal components analysis on a continous-valued matrix.</li>
<li>Projector  &#8212; Supplies a projector tool for tree windows that allows pictures attached to taxa to be displayed.</li>
<li>Proportion Gaps In Character  &#8212; Calculates the proportion of gaps (inapplicable codings) in a character across taxa.  Does not include missing (unassigned) data.</li>
<li>Proportion Gaps in Taxon  &#8212; Reports the proportion of gaps (inapplicable codings) in a taxon for a data matrix.</li>
<li>Proportion Internal Gaps  &#8212; Calculates the proportion of internal gaps (inapplicable codings) in a character across taxa.  Does not include missing (unassigned) data.</li>
<li>Proportion Invariant Model  &#8212; Defines and maintains models that specify that a certain proportion of characters are invariant.</li>
<li>Proportion lower case codings in taxon  &#8212; Reports the percentage of lower case codings in a taxon for a data matrix.</li>
<li>Proportion Missing  &#8212; Calculates the proportion of missing data in a character across taxa.</li>
<li>Proportion missing data in taxon  &#8212; Reports the proportion of missing data in a taxon for a data matrix.</li>
<li>Proportion Polymorphic  &#8212; Calculates the proportion of taxa coded as polymorphic or partially uncertain in a character.</li>
<li>Proportion Terminal Gaps  &#8212; Calculates the proportion of Terminal gaps (inapplicable codings) in a character across taxa.  Does not include missing (unassigned) data.</li>
<li>Protein Site Property  &#8212; Calculates the mean value of amino acid properties at a site across taxa.</li>
<li>Quick Key Entry  &#8212; Provides a tool with which to quickly enter data.  If this tool is active, then typing a key will cause that value to be entered into all selected cells.</li>
<li>Random Branch Moves  &#8212; Rearranges tree by random branch moves.</li>
<li>Random Fill (Categorical)  &#8212; Fills cells with a randomly-chosen state. For DNA data, states A, C, G, and T are chosen with equal probability; for other data, states up to and including the maximum state value are chosen with equal probability.</li>
<li>Random Fill (Gaussian/Normal)  &#8212; Fills cells with a random state, using a Normal distribution.</li>
<li>Random Fill (Meristic Uniform)  &#8212; Fills cells with a random state, uniformly.</li>
<li>Random Fill (Uniform)  &#8212; Fills cells with a random state, uniformly.</li>
<li>Random n characters  &#8212; Selects n characters randomly.</li>
<li>Randomize Taxon Order  &#8212; Randomizes the order of taxa.</li>
<li>Randomly Modify Current Tree  &#8212; Modifies current tree by random changes.</li>
<li>Randomly Modify Matrix  &#8212; Supplies character matrices that are randomly modified from an existing matrix.</li>
<li>Randomly Modify Respective Matrices  &#8212; Supplies character matrices that are randomly modified from a series of existing matrices.  The i'th modified matrix is derived from the i'th original matrix.</li>
<li>Randomly Modify Trees  &#8212; Modifies each of a series of trees by random changes; the i'th tree from this module comes by modifying the i'th tree from the original source of trees.</li>
<li>Randomly Resolve Polytomies  &#8212; Randomly resolves polytomies in tree.  All possible resolutions are equiprobable. Thus, if the tree is a polytomous bush, the resulting resolved trees will be distributed equivalently to that from the Equiprobable Trees module.</li>
<li>Rarefy Characters  &#8212; Deletes characters randomly to rarefy matrix.</li>
<li>Rarefy Tree  &#8212; Rarefies tree by randomly excluding taxa.  If some taxa are selected, random exclusion is limited to the selected taxa.</li>
<li>Raw Nexus block editor  &#8212; Edits blocks in a NEXUS file.</li>
<li>Read DATA blocks  &#8212; Coordinates the reading of a DATA block in NEXUS file.</li>
<li>Read DISTRIBUTION blocks  &#8212; Coordinates the reading of a DISTRIBUTION block in NEXUS file.</li>
<li>Rearranged tree  &#8212; Supplies trees that are rearrangments of a given tree.  The original tree is NOT included among the rearrangements.</li>
<li>Recode Characters  &#8212; Recodes categorical data (e.g., state 1 to state 0).</li>
<li>Reconstruct Ancestral States  &#8212; Reconstructs ancestral states on the nodes of a tree.</li>
<li>Reconstruct Deep Coalescence  &#8212; Reconstructs a contained tree within a containing tree so as to minimize the amount of deep coalescence (failure of lineage sorting).  The contained tree is assumed to be rooted.</li>
<li>Reinterpret Internal Node Labels...  &#8212; Reinterprets labels from all internal nodes as numbers or text attached to nodes or branches.</li>
<li>Remove from Taxon Names  &#8212; Removes a specified number of characters from taxon names.</li>
<li>Remove Gaps-Only Characters  &#8212; Removes all characters that are gaps only.</li>
<li>Remove Internal Node Labels  &#8212; Deletes labels from all internal nodes.</li>
<li>Remove Invariant Characters  &#8212; Removes all characters that have one state or fewer.</li>
<li>Remove item...  &#8212; Removes an item in each cell of a continuous matrix.</li>
<li>Remove Tokens &#8212; Removes tokens from start or end of the taxon name.</li>
<li>Rename item...  &#8212; Renames an item of a continuous matrix.</li>
<li>Replaces text in taxon names  &#8212; Replaces text in taxon names.</li>
<li>Reshuffle Character  &#8212; Supplies characters that are reshufflings of an existing character.</li>
<li>Reshuffle States Within Characters  &#8212; Shuffles (permutes) character states among taxa within each character.</li>
<li>Reshuffle States Within Taxa  &#8212; Shuffles (permutes) character states among characters within each taxon.</li>
<li>Reshuffle Terminal Taxa  &#8212; Shuffles (permutes) the taxa among the terminal nodes.</li>
<li>Reshuffle Within Characters (Taxa Partitioned)  &#8212; Shuffles (permutes) character states among taxa of each taxa partition, within each character.</li>
<li>Reshuffle Within Taxa (Char. Partitioned)  &#8212; Shuffles (permutes) character states among characters of each character partition, within each taxon.</li>
<li>Resolve Polytomies (to 0-length branches)  &#8212; Resolves polytomies arbitrarily and assigned the resulting new branches zero length.</li>
<li>Retention Index for Character  &#8212; Calculates the Retention Index (RI) for a tree and character.</li>
<li>Retention Index for Matrix   &#8212; Calculates the Retention Index (RI) for a tree and matrix.</li>
<li>Reverse Complement  &#8212; Reverses the DNA sequence and complements the bases.</li>
<li>Reverse Sequence  &#8212; Alters data by reversing sequence of states.</li>
<li>Rhetenor Package Introduction  &#8212; Multivariate analyses and morphometrics.</li>
<li>Right-Ladderize Tree  &#8212; Sets the tree to its ladderized (right) version.</li>
<li>Root tree  &#8212; Sets the tree to rooted.</li>
<li>Root tree with selected taxa as outgroup  &#8212; Roots the tree between the selected taxa and the remainder, if possible.</li>
<li>Rooting status  &#8212; Indicates whether the tree is rooted or unrooted.</li>
<li>s of Slatkin &amp; Maddison  &#8212; Calculates 's' of Slaktin &amp; Maddison 1989 for tree of genes associated with given populations</li>
<li>Sample Trees from Separate NEXUS File  &#8212; Supplies a fixed number of randomly-sampled trees directly from a file, without bringing the contained tree block entirely into memory.  This allows much larger blocks of trees to be used within constraints of memory, but will make some calculations slower.  This module does not know how many trees are in the file, and hence may attempt to read files beyond the number in the file.</li>
<li>Save matrix copies  &#8212; Provides for the saving of copies of matrices to separate files.  This is available under the Characters menu.</li>
<li>Scale All Branch Lengths  &#8212; Adjusts a tree's branch lengths by multiplying them by an amount.</li>
<li>Scale Selected Branch Lengths  &#8212; Adjusts lengths of a tree's selected branches by multiplying them by an amount.</li>
<li>Scattergram  &#8212; Helps make scattergram charts.</li>
<li>Scroll To Data  &#8212; Provides a tool that will scroll the data matrix to the next cell that contains data (i.e., that contains something other than a gap/inapplicable)</li>
<li>Search Data  &#8212; Manages data-searching modules.</li>
<li>Search for Better Tree  &#8212; Finds better trees (by the chosen optimality criterion) by rearranging the current tree.</li>
<li>Search within clade  &#8212; Supplies a tool for tree windows that searches for a better branch arrangement within clade touched.</li>
<li>Select Branches  &#8212; Provides a tool and menu with which to select branches in a tree window.</li>
<li>Select by Matrix Comparison  &#8212; Selects cells of the matrix that differ compared to another matrix</li>
<li>Select By Search (Find all)  &#8212; Selects cells of the matrix according to whether their text contains a given string</li>
<li>Select Lowercase Ends  &#8212; Selects ends of a DNA sequence up to the first block of N sites in a row with uppercase (confident) symbols (the user chooses N).  Assuming that lowercase letters are used for less certain base calls, this can be used to select and then trim (by painting with gaps) poorly-sequenced terminal regions.</li>
<li>Select Matching Taxa  &#8212; Selects taxa according to whether their sequence of states matches the sequence of states in the selected taxon</li>
<li>Select Same Distribution  &#8212; Selects cells of the matrix according to whether their distribution of states across taxa in a character is the same as that in the selected character or block of cells</li>
<li>Select Same Subsequence  &#8212; Selects cells of the matrix according to whether their sequence of states matches the sequence of states in the selected taxon or block of cells</li>
<li>Select Taxa  &#8212; Provides a tool with which to select taxa in a clade in a tree window.</li>
<li>Select Taxa from Name Search  &#8212; Select taxa based upon a name search.</li>
<li>Selected Block Mover  &#8212; Moves selected blocks of multiple sequences.</li>
<li>Selected Taxa Convex in Tree  &#8212; Determines if the selected taxa are convex in a tree.  That is, does there exist a rooting of the tree in which the selected taxa form a clade?</li>
<li>Selected Taxa Form Clade  &#8212; Determines if the selected taxa form a clade in a tree.</li>
<li>Selection Summary Strip &#8212; Shows the strip at bottom of matrix with character reference.</li>
<li>Semistrict Consensus  &#8212; Calculates the semistrict consensus tree.</li>
<li>Sequence Information Editor  &#8212; Edits character state names within the editor info panel.</li>
<li>Sequence Length  &#8212; Reports the length (total minus gaps) of a molecular sequence in a taxon.</li>
<li>Set Author  &#8212; Sets the author for this machine and account.</li>
<li>Set Polytomy Assumption to Default  &#8212; Sets assumption that any polytomies in the tree are whatever is current default (which can be set in the submenu File>Defaults>Tree Defaults).</li>
<li>Set Polytomy Assumption to Hard  &#8212; Sets assumption that any polytomies in the tree are hard.</li>
<li>Set Polytomy Assumption to Soft  &#8212; Sets assumption that any polytomies in the tree are soft.</li>
<li>Set to Lower Case  &#8212; Alters nucleotide data to lower case coding (e.g, to indicate less certain).</li>
<li>Set to Upper Case  &#8212; Alters nucleotide data to upper case coding (e.g, to indicate more certain).</li>
<li>Set Tree Defaults  &#8212; Sets the default state for polytomies &amp; other aspects of tree handling.</li>
<li>Shade numbers  &#8212; A module that displays numbers at tree nodes using labels.  This is a display-only module, and would be hired by another module that organizes assigning numbers to the nodes.</li>
<li>Shade states  &#8212; Shows the states at nodes by shading branches or nodes using colors, black and white, or shades of gray.</li>
<li>Shared Clades  &#8212; Calculates the number of shared clades between two trees (excludes the clade consisting of all taxa).</li>
<li>Shared History Distances (Covariance matrix) from Tree  &#8212; Distances among taxa implied by length of shared history (distance from root to most recent ancestor). This is the most common Covariance/Variance matrix used in GLS methods. Unassigned branches are treated as of length 1.0.</li>
<li>Shared Partitions  &#8212; Calculates the number of shared partitions between two trees.  If used as a distance, then converted by subtracting shared partitions from the maximum possible, i.e. the number of partitions in the subtree of shared taxa in the tree with the most partitions.</li>
<li>Shared Proportion Distances implied by Tree  &#8212; Distances among taxa implied by proportion of total branchlength (root to tip) that is shared (Shared Proportion distances).  Unassigned branches are treated as of length 1.0.</li>
<li>Shift Other To Match  &#8212; Shifts other sequences to match the one selected.</li>
<li>Shift to Minimize Stop Codons  &#8212; Shifts each sequence by 0, 1, or 2 bases to minimize number of stop codons implied.</li>
<li>Show Distance Matrix  &#8212; Writes distance matrix to log.</li>
<li>Show Employee Tree  &#8212; Shows the window listing the tree of employees of the module of a given window.</li>
<li>Show List with Genetic Codes  &#8212; Shows the List of Character window with the genetic codes column on.</li>
<li>Show Node Numbers  &#8212; Shows the node numbers on a tree.</li>
<li>Show Percentile  &#8212; Calculates and shows percentile boundaries.  Finds the upper (right) or lower (left) tail of the distribution that contains the specified percentage of the distribution. If there is not a boundary that exactly matches the specified percentage, then the closest boundary that does not exceed the percentage is shown.</li>
<li>Show text  &#8212; Shows text in a window</li>
<li>Show Y = X  &#8212; Shows the Y=X line on a scattergram</li>
<li>Shuffle states among taxa  &#8212; Alters data by shuffling states among taxa within a character.</li>
<li>Side Pusher  &#8212; Pushes all sequences on one side.</li>
<li>Simple (categorical data)  &#8212; Imports and exports simple matrices that consist of basic categorical data. Exported data will consist of default symbols (0, 1, 2, ...).</li>
<li>Simple (continuous data)  &#8212; Imports and exports simple matrices that consist of basic continuous data.</li>
<li>Simple (DNA/RNA)  &#8212; Imports and exports simple matrices that consist of DNA/RNA sequence data.</li>
<li>Simple (Geographic data)  &#8212; Imports and exports simple matrices that consist of basic geographic  data. The first character must be the latitude, and the second character the longitude.</li>
<li>Simple (protein)  &#8212; Imports and exports simple matrices that consist of amino acid sequence data.</li>
<li>Simple Tree Window  &#8212; Displays a single tree.</li>
<li>Simplicity Manager  &#8212; A small module to supervise the interface for simplification management.</li>
<li>Simplified NEXUS  &#8212; Exports NEXUS files, for instance of old fashioned style (using DATA block).</li>
<li>Simplify Taxon Names &#8212; Simplifies taxon names so that they will work with MrBayes, RAxML, etc..</li>
<li>Simulate Ancestral States  &#8212; Simulates ancestral states on the nodes of a tree.</li>
<li>Simulated Characters  &#8212; Supplies simulated characters.</li>
<li>Simulated Characters on Trees  &#8212; Supplies simulated characters each from a respective tree.</li>
<li>Simulated Matrices on Current Tree  &#8212; Supplies simulated character matrices.</li>
<li>Simulated Matrices on Trees  &#8212; Supplies simulated character matrices, each evolved on a different of a series of trees.</li>
<li>Simulated Tree Blocks  &#8212; Creates a tree block of simulated trees.</li>
<li>Simulated Trees  &#8212; Supplies trees from simulations.</li>
<li>Single Linkage  &#8212; Supplies trees obtained from Single Linkage clustering.</li>
<li>Sister Diversification  &#8212; Compares sister clades and returns the P value of the one-tailed null hypothesis that uniform clades of state 1 are smaller than uniform clades of state 0 among those with different values (P value calculated by Binomial probability).</li>
<li>Small State Names Editor  &#8212; Edits character state names within the editor info panel.</li>
<li>Sort Taxa (data)  &#8212; Provides a tool with which to sort taxa automatically.</li>
<li>Speciation/Extinction Likelihood  &#8212; Calculates likelihoods using a speciation/extinction model reduced from the BiSSE model (Maddison, Midford &amp; Otto 2007) </li>
<li>Speciation/extinction Likelihood Calculator  &#8212; Calculates likelihoods using a speciation/extinction model reduced from the BiSSE model (Maddison, Midford &amp; Otto 2007) </li>
<li>Splits multi-sequence block  &#8212; Splits blocks of multiple sequences.</li>
<li>SPR Rearranger  &#8212; Rearranges a tree by subtree pruning and regrafting (SPR).</li>
<li>Sprinkle Missing  &#8212; Randomly converts entries in the character matrix to missing data, with a certain probability (i.e., it "sprinkles" missing data around the matrix).</li>
<li>Sprinkle Missing in Cells  &#8212; Fills cells with a missing data with a specified probability.</li>
<li>Square tree  &#8212; Draws trees with standard square branches ("phenogram")</li>
<li>Standardize  &#8212; Alters continuous data by standardizing to have mean = 0 and unbiased estimate of variance = 1.  Modifies only the first item of a multi-item matrix</li>
<li>State Change Summarizer (over trees)  &#8212; Summarizes reconstructions of state changes of a character over a series of trees.</li>
<li>State Names Strip  &#8212; Shows the strip at bottom of matrix with state names.</li>
<li>States Information (in List of Characters window)  &#8212; Supplies basic character state information for characters in character list window.</li>
<li>Stepmatrix  &#8212; Supplies editor for and manages stepmatrices (cost matrices).</li>
<li>Steps in Character  &#8212; Calculates the parsimony steps in a character for a given tree.</li>
<li>Stochastic Character Evolution Package Introduction  &#8212; Provides stochastic models and simulations of character evolution.</li>
<li>Stochastic Character Mapping (Categorical)  &#8212; Coordinates reconstruction of ancestral states using stochastic character mapping. </li>
<li>Stored Character Order  &#8212; Supplies current order applied to characters for character list window.</li>
<li>Stored Characters  &#8212; Supplies characters from data files (as opposed to simulated characters, for example).</li>
<li>Stored Genetic Code  &#8212; Supplies a user-specified genetic code, stored in the file.</li>
<li>Stored Matrices  &#8212; Supplies character matrices from data files (as opposed to simulated characters, for example).</li>
<li>Stored Parsimony Model  &#8212; Supplies a user-specified model of character evolution, for parsimony calculations, stored in the file.  This can be different from the model assigned to the character as current in the List of Characters window.</li>
<li>Stored Probability Model  &#8212; Supplies a user-specified model of character evolution stored in the file.</li>
<li>Stored Probability Model for Simulation  &#8212; Supplies a user-specified model of character evolution stored in the file for simulation.</li>
<li>Stored Taxa  &#8212; Supplies taxa stored, for instance in a file.</li>
<li>Stored Taxa Associations  &#8212; Supplies associations between taxa that are stored, for instance in a file.</li>
<li>Stored Taxon Pairs  &#8212; Supplies taxon pairs stored, for instance in a file.</li>
<li>Stored Tree Blocks  &#8212; Supplies lists of trees stored, for instance in a file.</li>
<li>Stored Trees  &#8212; Supplies trees stored, for instance in a file.</li>
<li>Strict Consensus  &#8212; Calculates the strict consensus tree.</li>
<li>Sum of Branch Lengths  &#8212; Calculates the sum of branch lengths of the tree, treating unassigned lengths as 0.  The length of the root is counted.</li>
<li>Summarize Changes In Selected Clade  &#8212; Summarizes reconstructions of state changes of a character on descendants of selected branch, over a series of trees.  The branches defined are those that are the descendants of the single branch selected on the tree in the tree window.</li>
<li>Summarize State Changes Over Trees  &#8212; Summarizes reconstructions of state changes of a character over a series of trees.</li>
<li>Tab-delimited categorical data file  &#8212; Imports simple tab-delimited files of categorical data.</li>
<li>Tab-delimited continuous data file  &#8212; Imports and exports simple tab-delimited files of continuous data.</li>
<li>Taxa Association Package Introduction  &#8212; Provides utilities for managing taxa associations (e.g. for gene/species analyses).</li>
<li>Taxa blocks list  &#8212; Makes windows listing blocks of Taxa and information about them.</li>
<li>Taxa of data matrix  &#8212; Indicates taxa of data matrix.</li>
<li>Taxa of tree block  &#8212; Indicates taxa of tree block.</li>
<li>Taxa partition from Taxa associations  &#8212; Constructs a taxa partition from an association.</li>
<li>Taxa Scattergram  &#8212; Makes a scatterplot displaying two values (such as proportion of gaps in a sequence, etc.) for a series of taxa.</li>
<li>Taxon Group Color  &#8212; Colors by Taxon Group.</li>
<li>Taxon Has Data  &#8212; Indicates whether taxon has non-missing non-gap data in a given matrix.</li>
<li>Taxon Instability Among Trees  &#8212; Calculates for each taxon the degree to which its implied unweighted patristic distances between that taxon and others differs among trees.  For each taxon i this sums over all tree pairs x and y and over all other taxa j:  | Dijx - Dijy |/(Dijx + Dijy)^^z where Dijq is the distance between taxa i and j on tree q. Close relationships are emphasized if the exponent z is higher (default is 2).</li>
<li>Taxon Link  &#8212; Supplies a tool for tree windows that jumps to hypertext links for taxa.</li>
<li>Taxon List  &#8212; Makes windows listing taxa and information about them.</li>
<li>Taxon Pairs Histogram  &#8212; Makes a chart showing some value for each of many pairs of taxa.</li>
<li>Taxon Pairs Scattergram  &#8212; Makes a chart comparing two values for each of many pairs of taxa.</li>
<li>Taxon selection coordinator  &#8212; Coordinates taxon selection.</li>
<li>Taxon Wand (data)  &#8212; Provides a tool with which to select taxa automatically.</li>
<li>Terminal ? to Gaps  &#8212; Converts terminal missing data codings to gaps.</li>
<li>Terminal Gaps to ?  &#8212; Converts terminal gaps to missing data.</li>
<li>Text Window  &#8212; Displays Text in a window.</li>
<li>Ti/Tv Rate Matrix Model  &#8212; Defines and maintains two-parameter rate matrix models with a transition and transversion rates.</li>
<li>TNT  &#8212; Imports and exports TNT files.</li>
<li>Top BLAST Matches  &#8212; Does a BLAST search against GenBank on selected data and returns the top BLAST matches for each sequence selected.</li>
<li>Trace All Characters  &#8212; Summarizes for each node the reconstructions of the states at all characters of the tree.</li>
<li>Trace Character History  &#8212; Traces a history of character evolution on the nodes of a drawn tree.</li>
<li>Trace Character Over Trees  &#8212; Summarizes at each node reconstructions of the states of a character over a series of trees.  The summary is shown on the current tree; if you want to show it on a consensus of the trees, make sure that the current tree is that consensus.</li>
<li>Transform Trees from Other Source  &#8212; Transforms trees from another source.</li>
<li>Translate DNA to Protein  &#8212; Translates a stored DNA matrix to protein.</li>
<li>Tree Block Directly From File  &#8212; Supplies a block of trees directly from a file</li>
<li>Tree Block Values  &#8212; Makes a chart showing some value for each of many tree blocks.</li>
<li>Tree Blocks from Files Listed In File  &#8212; Supplies tree blocks in files listed in a file.</li>
<li>Tree Blocks List  &#8212; Makes windows listing tree blocks and information about them.</li>
<li>Tree Congruent with Constraint Tree Topology  &#8212; Determines if tree matches topology of a given constraint tree.  This module does not handle backbone constraints; all trees must have the same taxa present.  For backbone constraint trees, where the constraint tree need not contain all taxa, use the 'Tree Congruent with Backbone Constraint Tree Topology' module.</li>
<li>Tree Depth  &#8212; Calculates the deepest path (in branch length) from terminals to the root, treating unassigned lengths as 0.  The length of the root is not counted.</li>
<li>Tree Farm Package Introduction  &#8212; Extra utilities for trees, including comparisons, randomizations and manipulations.</li>
<li>Tree from ToL Web Project...  &#8212; Gets the tree for the page of the Tree of Life Web Project for the group specified.</li>
<li>Tree of context  &#8212; Supplies a single tree from the nearest tree context (e.g., an available tree window).</li>
<li>Tree of Life Web Project  &#8212; Provides a URL to a taxon on the Tree of Life Web Project</li>
<li>Tree of Life Web Project Interface Package Introduction  &#8212; Provides interfaces to the Tree of Life Web Project.</li>
<li>Tree Search  &#8212; Supplies trees resulting from a search to optimize some value.</li>
<li>Tree Search Package Introduction  &#8212; Coordinates simple searches for optimal trees.</li>
<li>Tree Utility Coordinator  &#8212; Coordinates use of tree utilities in tree window</li>
<li>Tree Value Satisfies Criterion  &#8212; Determines if the tree has a value either greater than, less than or equal to one specified.</li>
<li>Tree value using 2 characters  &#8212; Coordinates the calculation of a number for a tree based on 2 characters.</li>
<li>Tree value using character  &#8212; Coordinates the calculation of a number for a tree based on a character.</li>
<li>Tree value using character matrix  &#8212; Calculates a value for the tree using a character data matrix.</li>
<li>Tree Window  &#8212; Makes a basic tree window, which contains a tool palette.  Hires assistants for the tree window (e.g., Trace Character).</li>
<li>Tree Window Coordinator  &#8212; Coordinates the creation of basic tree windows.</li>
<li>TreeBase  &#8212; Provides a URL to a taxon in TreeBase</li>
<li>Treelength   &#8212; Calculates the parsimony treelength of a given tree and matrix.</li>
<li>Treelength  &#8212; Calculates the parsimony length of a tree.</li>
<li>Trees &amp; Diversification Characters  &#8212; Evolves a series of trees, each tied to a single character controlling diversification rates.</li>
<li>Trees List  &#8212; Makes windows listing trees and information about them.</li>
<li>Trees Package Introduction  &#8212; Basic management and utilities for phylogenetic trees.</li>
<li>Trees Scattergram  &#8212; Makes a scatterplot displaying two values (such as likelihoods, parsimony scores, imbalance statistics, correlations,etc.) for a series of trees.</li>
<li>Trim Terminal Gap Characters  &#8212; Trims characters at edges of matrix that are gaps only.</li>
<li>Trim Terminal Incomplete Codons  &#8212; Trims any incomplete codons from the start and end of each sequence.</li>
<li>Truncate taxon names  &#8212; Truncates taxon names.</li>
<li>Type of character data matrix  &#8212; Indicates type of character data matrix (e.g., continuous, DNA, etc.).</li>
<li>Type of Model  &#8212; Indicates type of model (e.g., Mk1, step-matrix) in list window.</li>
<li>uBio  &#8212; Provides a URL to a taxon in uBio</li>
<li>Unambiguous Changes  &#8212; Counts number of unambiguous changes along branch below based upon the character reconstruction.</li>
<li>Uncorrected (p) distance (DNA)  &#8212; Uncorrected (p) distance from a DNA matrix.</li>
<li>Uncorrected Distance (General)  &#8212; Uncorrected distance from a character matrix.</li>
<li>Uniform speciation (Yule)  &#8212; Generates tree by simple uniform probability speciation (a Yule process) as done by Harding (1971).  The chance of speciation is equal for all tips.</li>
<li>Uniform Speciation with Sampling  &#8212; Generates tree by simple uniform probability speciation (a Yule process), with subsequent sampling of species.</li>
<li>Unique ID (characters)  &#8212; Shows unique id assigned to character.</li>
<li>Unique ID (taxa)  &#8212; Shows unique id assigned to taxon.</li>
<li>Unique IDs default  &#8212; Sets the preferences for storing unique ids.</li>
<li>Unroot tree  &#8212; Sets the tree to unrooted.</li>
<li>UPGMA  &#8212; Supplies trees obtained from UPGMA clustering.</li>
<li>Use Trees from Separate NEXUS File  &#8212; Supplies trees directly from a file, without bringing the contained tree block entirely into memory.  This is a special purpose module designed to allow much larger blocks of trees to be used within constraints of memory, but will make some calculations slower.  Except for this special use, we recommend you use Include or Link from the file menu to access external tree files.  This module does NOT copy the trees into your main data file, and so if you save your main data file then move it or the tree file, the data file will no longer be able to find the trees.  This module does not know how many trees are in the file, and hence may attempt to read files beyond the number in the file.</li>
<li>User-Specified Nucleotide Frequency Model  &#8212; Defines and maintains user-specified nucleotide frequency models.</li>
<li>Utilities  &#8212; Provides a menu for utilities</li>
<li>Value for 2 characters with current tree  &#8212; Coordinates the calculation of a number for two characters based on a current tree.</li>
<li>Value for Clade of Selected Taxa  &#8212; Coordinates the calculation of a value for the clade of selected terminal taxa.  If no terminal taxa are selected, the value at the root is given.  If the selected taxa do not make a clade, no answer is returned.</li>
<li>Value for Node Selected In Current Tree  &#8212; Coordinates the calculation of a value for a node selected in a current tree.  If no node or more than one node is selected in current tree, no result is returned.  If the assessed tree does not share the same clade of the selected node, no answer is returned.</li>
<li>Value from Tree  &#8212; Calculates a value for each taxon using a tree.</li>
<li>Values (Tree information panel)...  &#8212; Makes the Values section of the Tree Information Panel to display values pertaining to the tree.</li>
<li>Values for Current Tree  &#8212; Makes the legend in a tree window to display the tree name and values pertaining to the tree.</li>
<li>Values for Nodes  &#8212; Shows on a drawn tree various possible numbers at the nodes.</li>
<li>Variable (Mult. States)  &#8212; Indicates whether a character has multiple observed states.  If taxa are selected, only the selected taxa are examined.</li>
<li>Variable among taxa  &#8212; Selects characters that are variable among the selected taxa</li>
<li>Visibility of Matrix  &#8212; Indicates whether character data matrix is hidden or not.</li>
<li>Web page link  &#8212; Provides a window to jump to a web page</li>
<li>Whole or Submodel  &#8212; Indicates whether model is a complete or partial model of character evolution.</li>
<li>Window Holder  &#8212; Helps other modules by holding their windows.</li>
<li>With probability p  &#8212; Selects characters randomly with probability p.</li>
<li>Within-group PCA  &#8212; Performs within-group principal components analysis on a continous-valued matrix.</li>
              </ul>
              <!-- #EndEditable --></font></td>
          </tr>
        </table>
        <table border="0" cellspacing="0" cellpadding="5" width="100%">
          <tr> 
            <td align="left" valign="top"> 
              <hr>
              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite
              Home Page</font></a><font size="2"> | <a href="manual.html">Mesquite
              Manual</a></font></span></font></p>
              <p><font face="Trebuchet MS"><span class="smallplain">Copyright 
                &copy; 2002-2010 by <a href="http://salticidae.org/wpm/home.html">Wayne 
                P. Maddison</a> and <a href="http://david.bembidion.org">David 
                R. Maddison</a>.<br>
&nbsp;All rights reserved.</span></font></p></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</div>

<font face="Trebuchet MS"> 
</font>
</body>
<!-- #EndTemplate --></html>
