<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 -->
<head>
<!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"-->
<!-- #BeginEditable "doctitle" --> <title>Are strepsipterans related to flies? Exploring long branch attraction</title><!-- #EndEditable -->
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body bgcolor="#FFFFFF">
<div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%"> 
  <table border="0" cellspacing="0" cellpadding="0" width="100%">
    <tr> 
	  <td align="left" valign="top" height="60"> 
        <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">
          <tr> 
       		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="../../images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>
           <td bgcolor="#FFFFFF"> 
              <div align="left"><font face="Trebuchet MS"><img src="../../images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></font></div>
            </td>
          </tr>
			<tr> 
			  <td height="42" align="left" valign="top"></td>
			  <td height="42" align="left" valign="top"><font face="Trebuchet MS"><img src="../../images/mesquiteLower.gif"></font></td>
			</tr>
        </table>
        <a href="../../manual.html"><img src="../../images/manual.jpg" width="100" height="29" border="0"></a></td>
	<td>
        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">
          <tr valign="top" height="92"> 
           	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="../../images/fadeRight.gif" width="72" height="160"></font></td>
			<td bgcolor="#99FF99"><img src="../../images/introduction.gif" valign="bottom"></td>
			<td align="center" valign="top" height="92" width = "160" bgcolor="#99FF99">
		  		<div align="left"> 
                <p><font face="Trebuchet MS"><span class="smallplain"><a href="../../manual.html"><br>
                  Introduction<br>
                  </a></span><a href="../../whyMesquite.html">Why?</a><br>
                  <a href="../../publication.html">Publication</a><br>
                  <a href="../../support.html">Support</a><br>
                  <a href="../../credits.html">Credits</a> </font><br>
                  <font face="Trebuchet MS"><span class="smallplain"><a href="../../help.html">Help 
                  </a></span></font><br>
                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web Site</font></a><br>
				  <a href="../../toc.html"><img src="../../images/toc.gif" width="80" height="40" border="0" align="bottom"></a></p>
              </div>
			</td>
      		<td bgcolor="#99FF99"><img src="../../images/basics.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99"> 
              <p><font face="Trebuchet MS"><span class="smallplain"><a href="../../help.html"><br>
                </a><a href="../../managingFiles.html">Files</a></span> <br>
                <span class="smallplain"><a href="../../menus.html">Menus</a><br>
                <a href="../../windows.html">Windows</a><br>
				<a href="../../charts.html">Charts</a><br>
                <a href="../../scripting.html">Scripts/Macros</a> <br>
                <a href="../../modules.html">Modules</a><br>
				<a href="../../howMesquiteWorks.html">How</a></span></font>
				<!-- Google CSE Search Box Begins  -->
				  <form id="searchbox_016202157087875381635:khljhzi-ps8" action="http://www.google.com/cse">
				    <input type="hidden" name="cx" value="016202157087875381635:khljhzi-ps8" />
				    <input type="hidden" name="cof" value="FORID:0" />
				    <input name="q" type="text" size="20" />
				    <input type="submit" name="sa" value="Search Mesquite" />
				  </form>
				  <script type="text/javascript" src="http://www.google.com/coop/cse/brand?form=searchbox_016202157087875381635%3Akhljhzi-ps8"></script>
				<!-- Google CSE Search Box Ends --></p>
              </td>
      		<td bgcolor="#99FF99"><img src="../../images/objects.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">
				<p><font face="Trebuchet MS"><br>
                <span class="smallplain"><a href="../../characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="../../Taxa.html"><br>
                Taxa<br>
                </a></span></font><font face="Trebuchet MS"><a href="../../trees.html">Trees</a><br>
                <a href="../../glossary.html">Glossary</a> </font></p>
              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="../../changes.html">New</a><br>
                &nbsp;&nbsp;<a href="../../changes.html">Features</a></font></em></font></p>
              </td>
      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="../../images/fadeLeft.gif" width="92" height="160"></font></td>
		</tr>
    	<tr>
		    <td width="72" height="20"></td>
		<td></td>
		    <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td>&nbsp;</td>
		</tr>
		</table>
		</td>
	</tr>
  </table>
  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">
    <tr> 
      <td valign="top" align="left" width="110" height="100%"> 
        <table border="0" cellspacing="0" cellpadding="0" width="142">
          <tr> 
            <td width="6" bgcolor="#99FF99"><font face="Trebuchet MS"><img src="../../images/fadeDown.gif" width="5" valign = "top" height="95" border="0" alt=" "></font></td>
            <td colspan="2" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="../../images/fadeDown.gif" width="100%" height="72" border="0" alt=" "><img src="../../images/analyses.gif"></font></p>
              </td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>
            <td width="131" height="35" align="left" valign="middle"><p>
			  <font face="Trebuchet MS">
			  <a href="../../CharacterEvolution/charEvolution.html">Character</a><br>
			    &nbsp; <a href="../../CharacterEvolution/charEvolution.html">Evolution</a> <br> <br>
			  <a href="../../Diversification/diversification.html">Diversification</a><br><br>
			  <a href="../../molecular/molecular.html">Molecular</a><br>
			    &nbsp; <a href="../../molecular/molecular.html">Data</a><br><br>
			  <a href="../../popGen/popGen.html">Gene trees</a> <br>  <br>
			  <a href="../../RandomSimul/RandomSimul.html">Simulations &amp;</a><br>
			    &nbsp; <a href="../../RandomSimul/RandomSimul.html">Randomizations</a> <br> <br>
			  <a href="../../continuous/continuous.html">Continuous</a><br>
			    &nbsp; <a href="../../continuous/continuous.html">Characters</a><br> <br>
			  <a href="../../otherPrograms.html">Use with</a><br>
			    &nbsp; <a href="../../otherPrograms.html">Other Programs</a> <br><br>
			  <a href="../index.html">Studies</a> </font></p></td>
            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr> 
            <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="../../images/fadeUp.gif" width="5" height="92"></font></td>
            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="../../images/fadeUp.gif" width="100%" height="92"></font></td>
          </tr>
        </table>
      </td>
      <td valign="top" width="100%" height="100%"> 
        <table width="700" border="0" cellpadding="10" cellspacing="0">
          <tr> 
            <td><font face="Trebuchet MS"><!-- #BeginEditable "body" -->               <h2>Are strepsipterans related to flies? Exploring long branch attraction</h2>              <p>David R. Maddison</p>              <h3 align="left">The question</h3>              <p align="left"><a href="http://tolweb.org/tree?group=Strepsiptera">Strepsiptera</a>,                sometimes known as twisted-wing parasites, is an enigmatic group                of insects. They are parasites of other insects. The males have                only one pair of wings and  odd raspberry-like eyes and                lobed antennae. Females are wingless, and in most species never                leave the host. These insects have traditionally been considered                related to <a href="http://tolweb.org/tree?group=Coleoptera">beetles</a>,                although that placement is supported by little evidence. Recently,                molecular data have suggested that they may instead be related                to true, two-winged flies (<a href="http://tolweb.org/tree?group=Diptera">Diptera</a>). </p>              <p align="left">For example, parsimony analysis of a small data                matrix of 18S ribosomal DNA yields (Carmean and Crespi, 1995)                a phylogeny in which Strepisptera is placed as sister group of                Diptera (in yellow) rather than Coleoptera (in orange):</p>              <p align="center"><img src="images/parsimony.gif" width="227" height="273"></p>              <p align="left">This is the strict consensus tree of 27 most-parsimonious                trees.</p>              <p align="left">However, examining the relative branch lengths              for any one of the trees, it becomes evident that the branches                for Diptera and Strepsiptera are unusually long:</p>              <p align="center"><img src="images/parsimonyLB.gif" width="287" height="273"></p>              <p align="left">This raised the possibility that the association                between Strepsiptera and Diptera in these trees was an artifact,                caused by long-branch attraction (Felsenstein, 1978; Hendy and                Penny, 1989).</p>              <p align="left">Huelsenbeck (1997) set to determine if the relationship                seen in this analysis of Strepsiptera with Diptera could be accounted                for by long-branch attraction. To do this, he                conducted                a simulation                study. We won't reproduce his whole study here, just one part              of his Figure 2. </p>              <p align="left">We will ask only one simple question: If strepsipterans                  are indeed related to beetles, would our observation that parsimony                  analysis yields a phylogeny with strepsipterans related to                Diptera be unexpected? If we can show that it is unexpected,                we can reject                  the notion that strepsipterans are related to beetles. If,                however, inference of strepsipterans as related to Diptera is                expected even if they are actually related to beetles, then we                cannot so readily reject the traditional view.</p>              <h3 align="left">A simulation study</h3>              <p align="left">To conduct a simulation study, we first need a                model tree. We can use a tree inferred using maximum likelihood,                which places strepsipterans with beetles:</p>              <p align="center"><img src="images/modelTreePlain.gif" width="275" height="274"></p>              <p align="center"><em>Model Tree</em></p>              <p align="left">The branch lengths of this tree were inferred using                maximum likelihood. We also need a full model of DNA sequence                evolution (which can also be inferred using maximum likelihood).</p>              <p align="left">With this model in hand, we can simulated the evolution                of 18S rDNA up the branches of this phylogeny, to yield a simulated                matrix. We can then infer the phylogeny for this simulated matrix                using parsimony, and see where Strepsiptera falls. Repeating                this multiple times will give us an idea about the expected placement                of Strepsiptera using parsimony inference presuming the model                tree shown above.</p>              <h3 align="left">Conducting the simulations</h3>              <p align="left">To conduct the simulations, in Mesquite open the                example file &quot;<strong>study002.nex</strong>&quot;, in the                folder Mesquite_Folder/docs/mesquite/studies/study002/. This                file contains the 13-taxa data matrix from Carmean and Crespi                (1995) as modified by Huelsenbeck (1977), as well as the model                tree shown above. The model tree will appear in a window when                you open the file:</p>              <p align="center"><img src="images/modelTree.gif" width="388" height="380"></p>              <p align="left">In addition, a model of character evolution has                been entered into this file, with parameters obtained from maximum                likelihood inference using PAUP*4 (Swofford, 2003). These were                inferred on the model tree using the observed 18S rDNA data.                </p>              <p align="left">You can see the model by choosing <u><font size="-1" face="Arial, Helvetica, sans-serif">Characters&gt;Edit                  Character Model&gt;Huelsenbeck Model</font></u>:</p>              <p align="center"><img src="images/model.gif" width="363" height="314"></p>              <p align="left">This model uses the empirical state frequencies                as present in the original 18S rDNA matrix, a model of character                rate variation called &quot;gammaModel&quot;, and a rate matrix model called                &quot;HKY85&quot;. You can see the nature of gammaModel by touching on                the Edit button near its listing:</p>              <p align="center"><img src="images/gamma.gif" width="214" height="151"></p>              <p align="left">and the nature of the HKY85 model by touching on                its Edit button:</p>              <p align="center"><img src="images/titv.gif" width="275" height="108"></p>              <p align="left">With the model fully established, we can now conduct                the simulation study. To do this, we want to ask Mesquite to                create many simulated matrices, each evolved according to the                model. We also want to create a script (&quot;batch file&quot;)                that will tell a tree-inference program (we will use PAUP*, but                another program such as NONA could be used) to find the most                parsimonious trees for each of the matrices, and summarize the                results. Mesquite's Batch Architect package contains the tools                to automate this process.</p>              <p align="left">Choose <u><font size="-1" face="Arial, Helvetica, sans-serif">Analysis&gt;Batch                  Architect &gt; Export                  Matrices &amp; Batch                  Files...</font></u> In the                  first dialog box that appears choose Simulated Matrices on                  Current Tree, then Evolve DNA Characters, then Huelsenbeck                  Model. You will be presented with the Export Matrices &amp; Batch                  Files dialog box, in which you can enter the base name for                  the matrix files to be created and the number of matrices (100                  is a good start).</p>              <p align="center"><img src="images/embf.gif" width="336" height="356"></p>              <p align="left">The batch file template to be used is the one called                &quot;Basic PAUP tree search&quot;. This template builds a PAUP* command                file that will tell PAUP to execute and analyze each matrix in                turn, and then, and the end, harvest the results and calculate                a majority-rule consensus tree.</p>              <p align="left">After pressing OK, you will be asked for a location                to save the 100 matrices. It is recommended that you have an                empty folder available into which they can be saved to avoid                cluttering up another folder with many files. You will be asked                one last question: the number of characters to be evolved in                each matrix. As we want the simulation model to be as similar                as possible to reality, we will chose to evolve the same number                of characters as is present in the observed matrix, which is                770.</p>              <p align="left">Mesquite will now simulate the matrices, and produce                a batch file called &quot;paupCommands.nex&quot;. This file consists of                commands for PAUP*. The start of the file looks something like                this:</p>              <p align="center"><img src="images/paupStart.gif" width="433" height="209"></p>              <p align="left">After some initial setup, PAUP* is to execute the                first simulated matrix file (&quot;StrepSim0.nex&quot;), do a                heuristic search (which, by default, should be for most-parsmonious                trees),                and then save the results to a tree file. It then saves the strict                consensus                tree                of the                most-parsimonious                trees                to a different tree file. It repeats this process for the next                matrix, StrepSim1.nex, and the next one, StrepSim2.nex, and so                on, until it gets to StrepSim99.nex, as requested near the end                of                the file:</p>              <p align="center"><img src="images/paupEnd.gif" width="431" height="124"></p>              <p align="left">After processing StrepSim99.nex, it reads in the                tree file containing the accumulated consensus trees, and calculates                both a strict consensus tree of the results from the 100 replicates,                as well as the majority rule consensus tree of the replicate's                results.</p>              <p align="left">To ask PAUP* to do this analysis, open PAUP* and                ask it to execute the file paupCommands.nex.</p>              <h3>Interpreting the results</h3>              <p>After PAUP* finished executing paupCommands.nex, near the bottom                of PAUP*'s main window will be the majority rule consensus tree                of the strict                consensus                trees                from                each                of the 100 replicates. It will look something like this:</p>              <p align="center"><img src="images/result.gif" width="547" height="174"></p>              <p>The actual numbers may differ, but they should be similar to                those shown.</p>              <p>The &quot;73&quot; on the Strepsiptera + Aedes + Drosophila                branch indicates that in 73 of the simulated matrices, the most                parsimonious trees                had Strepsiptera with Diptera, even though the model tree on                which the data were simulated had Strepsiptera with beetles.                This indicates that the inferred placement of strepsipterans                with dipterans with the observed data might be the result of                an inference artifact, as inferring strepsipterans with Diptera                is expected even if they truly are with beetles.</p>              <h3>Summary</h3>              <font face="Trebuchet MS">              <p>Simulations were done to see what trees we would expect from                a phylogeny inference under a particular model tree. The steps                in the study are:</p>              <p>               <ul>                <li>The branch lengths of the model tree are inferred using maximum                likelihood in PAUP*. </li>                <li> Values of parameters of a model of sequence evolution (gamma                  shape parameter, transition/transversion rate) are inferred                  using maximum likelihood in PAUP* on the model tree using the                  observed DNA sequences.</li>                <li>The data matrix is opened in Mesquite, along with the model                   tree. Submodels and a model of character evolution are created                   within Mesquite to match those inferred.</li>                <li>Mesquite's Batch Architect is used to automate the process                   of simulating the evolution of multiple (100 or more) data matrices                   under this model, using Mesquite's Genesis package. Batch Architect                   also builds a command file for PAUP* and an instruction file                   for Mesquite so that it can interpret the results of the PAUP*                   analyses.</li>                <li>The command file is executed in PAUP*, telling PAUP* to search                  for the shortest trees for each                  of the simulated matrices, accumulating the consensus trees                  for each matrix in a tree file. PAUP* is also instructed to                  calculate a majority-rule consensus tree of the results of                  each analysis.</li>                <li>The majority-rule consensus tree is examined in PAUP* to                  see what trees are expected to be inferred under these conditions.<font face="Trebuchet MS"> </font>                   </li>              </ul>              </font><font face="Trebuchet MS"><h3>References</h3>              <p>Carmean, D., and B. Crespi. 1995. Do long branches attract flies?                Nature, 373:666.</p>              <p>Felsenstein, J. 1978. Cases in which parsimony                  and compatibility methods                  will                  be positively                  misleading.                  Systematic                  Zoology, 27,                                  401-410.</p>              <p>Hendy, M.D., and Penny, D. 1989. A framework for the quantitative                 study of evolutionary trees. Systematic Zoology, 38, 297-309.</p>              <p>Huelsenbeck, J.P. 1997. Is the Felsenstein zone a fly trap?                 Systematic Biology, 46, 69-74.</p>              <p>Swofford, D. L. 2003. PAUP*. Phylogenetic Analysis Using Parsimony                (*and Other Methods). Version 4 beta 10. Sinauer Associates,              Sunderland, Massachusetts. </p>              <p>&nbsp;</p>              <p>              <HR></p>              <h4>Citation for this page</h4>              <p><font size="-1">Maddison, D.R. 2004. Are strepsipterans related                 to flies? Exploring long branch attraction. Study 2 <em>in</em>                 Mesquite: a modular system for evolutionary analysis, version<!-- #BeginLibraryItem "/Library/MesquiteVersionNumber.lbi" --> 1.0<!-- #EndLibraryItem -->4,                 http://mesquiteproject.org.</font></p>              <p>&nbsp;</p>            </font> <!-- #EndEditable --></font></td>
          </tr>
        </table>
        <table border="0" cellspacing="0" cellpadding="5" width="100%">
          <tr> 
            <td align="left" valign="top"> 
              <hr>
              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite
              Home Page</font></a><font size="2"> | <a href="../../manual.html">Mesquite
              Manual</a></font></span></font></p>
              <p><font face="Trebuchet MS"><span class="smallplain">Copyright 
                &copy; 2002-2006 by <a href="http://salticidae.org/wpm/home.html">Wayne 
                P. Maddison</a> and <a href="http://david.bembidion.org">David 
                R. Maddison</a>.<br>
&nbsp;All rights reserved.</span></font></p></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</div>

<font face="Trebuchet MS"> 
</font>
</body>
<!-- #EndTemplate --></html>
